<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Exporting Artifacts | Bacterial 16S metabarcoding supplemental</title>
  <meta name="description" content="Chapter 2 Exporting Artifacts | Bacterial 16S metabarcoding supplemental" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Exporting Artifacts | Bacterial 16S metabarcoding supplemental" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Exporting Artifacts | Bacterial 16S metabarcoding supplemental" />
  
  
  

<meta name="author" content="Matthew Gemmell" />


<meta name="date" content="2021-03-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="01-16S_supplemental.html"/>
<link rel="next" href="03-Functional_prediction.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a>Illumina QC</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-16S_supplemental.html"><a href="01-16S_supplemental.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html"><i class="fa fa-check"></i><b>2</b> Exporting Artifacts</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#biom-files"><i class="fa fa-check"></i><b>2.1</b> BIOM files</a></li>
<li class="chapter" data-level="2.2" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#setup"><i class="fa fa-check"></i><b>2.2</b> Setup</a></li>
<li class="chapter" data-level="2.3" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#abundance-table-export"><i class="fa fa-check"></i><b>2.3</b> Abundance table export</a></li>
<li class="chapter" data-level="2.4" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#taxonomy-info-export"><i class="fa fa-check"></i><b>2.4</b> Taxonomy info export</a></li>
<li class="chapter" data-level="2.5" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#representative-sequences-export"><i class="fa fa-check"></i><b>2.5</b> Representative sequences export</a></li>
<li class="chapter" data-level="2.6" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#phylogenetic-tree-export"><i class="fa fa-check"></i><b>2.6</b> Phylogenetic tree export</a></li>
<li class="chapter" data-level="2.7" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#export-asv-table-with-taxonomy"><i class="fa fa-check"></i><b>2.7</b> Export ASV table with taxonomy</a></li>
<li class="chapter" data-level="2.8" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#export-rarified-table"><i class="fa fa-check"></i><b>2.8</b> Export rarified table</a></li>
<li class="chapter" data-level="2.9" data-path="02-Exporting_artifacts.html"><a href="02-Exporting_artifacts.html#export-genus-table"><i class="fa fa-check"></i><b>2.9</b> Export genus table</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Functional_prediction.html"><a href="03-Functional_prediction.html"><i class="fa fa-check"></i><b>3</b> Functional prediction</a></li>
<li class="chapter" data-level="4" data-path="04-16S_R_packages.html"><a href="04-16S_R_packages.html"><i class="fa fa-check"></i><b>4</b> 16S R packages</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bacterial 16S metabarcoding supplemental</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="exporting-artifacts" class="section level1" number="2">
<h1><span class="header-section-number">Chapter 2</span> Exporting Artifacts</h1>
<p><img src="figures/artefact_to_tsv.png" width="20%" style="display: block; margin: auto;" /></p>
<p>Artifact files are very useful when carrying out analysis in QIIME2. However they are impossible to use (I think) with anything outside of QIIME2. Therefore it can be very useful to know how to export data out of Artifact files.</p>
<p>This section will show how to export various different artifact files into text files. However, for some data types they must be converted into a BIOM file first.</p>
<div id="biom-files" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> BIOM files</h2>
<p><img src="figures/biom.png" width="20%" style="display: block; margin: auto;" />
BIOM stands for BIological Observation Matrix. BIOM files generally contain abundance tables. These files are not human-readable but can be used with certain programs. Here we will only manipulate them in such ways to get the text files we need.</p>
<p>For more information on the BIOM format please see: <a href="https://biom-format.org/" class="uri">https://biom-format.org/</a></p>
<p>We will be using the command <code>biom convert</code> to convert BIOM files to TSV (Tab Separated Value) files. For more info on this please see: <a href="https://biom-format.org/documentation/biom_conversion.html" class="uri">https://biom-format.org/documentation/biom_conversion.html</a></p>
</div>
<div id="setup" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Setup</h2>
<p><img src="figures/setup.png" width="20%" style="display: block; margin: auto;" />
First copy a new directory that will contain all the QIIME2 output you need and then change directory to it.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="02-Exporting_artifacts.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> -r /pub39/tea/matthew/NEOF/16s_workshop/16s_export ~</span>
<span id="cb1-2"><a href="02-Exporting_artifacts.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ~/16s_export</span></code></pre></div>
<p>Before we start exporting it is best to make a few directories.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="02-Exporting_artifacts.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> export</span>
<span id="cb2-2"><a href="02-Exporting_artifacts.html#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> qiime2_exports</span>
<span id="cb2-3"><a href="02-Exporting_artifacts.html#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> final_exports</span></code></pre></div>
<ul>
<li><code>export</code>: This directory is where we will point the various <code>qiime tools export</code> commands. You can only specify an output directory for this command and the command will give the exported file very generic names. This makes it very easy to overwrite files when using this command.</li>
<li><code>qiime2_exports</code>: For some of the exported files we will move and rename them from <code>export</code> to this directory. This is an temporary directory for files we will not want at the end (mostly BIOM files).</li>
<li><code>final_exports</code>: This directory will contain the final exported files we would generaly want to keep.</li>
</ul>
</div>
<div id="abundance-table-export" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Abundance table export</h2>
<p>First we will export our ASV abundance table. We will have to convert the BIOM file produced by QIIME2 to a TSV file.</p>
<p>Export the table to a directory. This will create a BIOM file called <code>feature-table.biom</code>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="02-Exporting_artifacts.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools export --input-path table-dada2.qza --output-path export</span></code></pre></div>
<p>Move the created file whilst renaming it.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="02-Exporting_artifacts.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> export/feature-table.biom qiime2_exports/ASV_table.biom</span></code></pre></div>
<p>Finall we will use <code>biom convert</code> to create our TSV file.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="02-Exporting_artifacts.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="ex">biom</span> convert <span class="kw">\</span></span>
<span id="cb5-2"><a href="02-Exporting_artifacts.html#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="ex">-i</span> qiime2_exports/ASV_table.biom <span class="kw">\</span></span>
<span id="cb5-3"><a href="02-Exporting_artifacts.html#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="ex">-o</span> final_exports/ASV_table.tsv <span class="kw">\</span></span>
<span id="cb5-4"><a href="02-Exporting_artifacts.html#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="ex">--to-tsv</span></span></code></pre></div>
<p>The option <code>--to-tsv</code> (or alternative) must be provided or the command will not run and only complain.</p>
<p>Use a text viewer/editor of your choice to inspect the final file. Make sure to do this for all the files in this tutorial.</p>
</div>
<div id="taxonomy-info-export" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Taxonomy info export</h2>
<p>Exporting the taxonomy info.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="02-Exporting_artifacts.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tool export --input-path taxonomy.sklearn.qza <span class="kw">\</span></span>
<span id="cb6-2"><a href="02-Exporting_artifacts.html#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="ex">--output-path</span> export</span>
<span id="cb6-3"><a href="02-Exporting_artifacts.html#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Move the produced tsv file to the final_exports directory</span></span>
<span id="cb6-4"><a href="02-Exporting_artifacts.html#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> export/taxonomy.tsv final_exports</span></code></pre></div>
</div>
<div id="representative-sequences-export" class="section level2" number="2.5">
<h2><span class="header-section-number">2.5</span> Representative sequences export</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="02-Exporting_artifacts.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools export --input-path rep-seqs-dada2.qza <span class="kw">\</span></span>
<span id="cb7-2"><a href="02-Exporting_artifacts.html#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="ex">--output-path</span> export</span>
<span id="cb7-3"><a href="02-Exporting_artifacts.html#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Move and rename the produced fasta file</span></span>
<span id="cb7-4"><a href="02-Exporting_artifacts.html#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> export/dna-sequences.fasta final_exports/ASV_rep_seqs.fasta</span></code></pre></div>
</div>
<div id="phylogenetic-tree-export" class="section level2" number="2.6">
<h2><span class="header-section-number">2.6</span> Phylogenetic tree export</h2>
<p>The below export the specified phylogenetic tree into a newick format file. For more info on the Newick format please see: <a href="https://en.wikipedia.org/wiki/Newick_format" class="uri">https://en.wikipedia.org/wiki/Newick_format</a></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="02-Exporting_artifacts.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools export --input-path rooted-tree.qza <span class="kw">\</span></span>
<span id="cb8-2"><a href="02-Exporting_artifacts.html#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="ex">--output-path</span> export</span>
<span id="cb8-3"><a href="02-Exporting_artifacts.html#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Move and rename the exported newick file</span></span>
<span id="cb8-4"><a href="02-Exporting_artifacts.html#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> export/tree.nwk final_exports/fasttree_rooted_tree.nwk</span></code></pre></div>
</div>
<div id="export-asv-table-with-taxonomy" class="section level2" number="2.7">
<h2><span class="header-section-number">2.7</span> Export ASV table with taxonomy</h2>
<p>This step will produce a TSV file containing the ASV abundance table with the last column containing the taxonomy assignment of the ASV.</p>
<p>To carry out this step you will need the BIOM format produced in the abundance table export and the exported taxonomy file.</p>
<p>First we create a new BIOM file which contains the taxonomy information</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="02-Exporting_artifacts.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">biom</span> add-metadata <span class="kw">\</span></span>
<span id="cb9-2"><a href="02-Exporting_artifacts.html#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="ex">--input-fp</span> qiime2_exports/ASV_table.biom <span class="kw">\</span></span>
<span id="cb9-3"><a href="02-Exporting_artifacts.html#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="ex">--observation-metadata-fp</span> export/taxonomy.tsv <span class="kw">\</span></span>
<span id="cb9-4"><a href="02-Exporting_artifacts.html#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="ex">--output-fp</span> qiime2_exports/ASV_table.tax.biom</span></code></pre></div>
<p>Now we export the newly created BIOM file. We will need to include option <code>--header-key taxonomy</code> to include the taxonomy info in the newly TSV file.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="02-Exporting_artifacts.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="ex">biom</span> convert <span class="kw">\</span></span>
<span id="cb10-2"><a href="02-Exporting_artifacts.html#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="ex">-i</span> qiime2_exports/ASV_table.tax.biom <span class="kw">\</span></span>
<span id="cb10-3"><a href="02-Exporting_artifacts.html#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="ex">-o</span> final_exports/ASV_table.tax.tsv <span class="kw">\</span></span>
<span id="cb10-4"><a href="02-Exporting_artifacts.html#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="ex">--header-key</span> taxonomy <span class="kw">\</span></span>
<span id="cb10-5"><a href="02-Exporting_artifacts.html#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="ex">--to-tsv</span></span></code></pre></div>
</div>
<div id="export-rarified-table" class="section level2" number="2.8">
<h2><span class="header-section-number">2.8</span> Export rarified table</h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="02-Exporting_artifacts.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> feature-table rarefy --i-table table-dada2.qza --p-sampling-depth 56844 --o-rarefied-table table-dada2.rarefied_56844.qza</span></code></pre></div>
</div>
<div id="export-genus-table" class="section level2" number="2.9">
<h2><span class="header-section-number">2.9</span> Export genus table</h2>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb12-1"><a href="02-Exporting_artifacts.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> taxa collapse --i-table ../table-dada2.rarefied_56844.qza --i-taxonomy ../taxonomy.sklearn.qza --p-level 6 --o-collapsed-table table-dada2-order.rarefied_56844.qza</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="01-16S_supplemental.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="03-Functional_prediction.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
