# De-novo amplicon sequence variants identification
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/clusters.png", auto_pdf = TRUE)
``` 

We will use DADA2 to identify the amplicon sequence variants (ASVs) among all the samples. 

## DADA2 denoising

To perform the variant sequence identification step with DADA2, the command is (don't run this command, see below command):

```{bash, eval=FALSE}
qiime dada2 denoise-paired \
--i-demultiplexed-seqs paired-end-demux.trim.qza \
# Truncates R1 sequences after the 220 nt length
--p-trunc-len-f 220 \
# Truncates R2 sequences after the 220 nt length
--p-trunc-len-r 220 \
# Object containing the identified variants
--o-representative-sequences rep-seqs-dada2.qza \
# Object containing abundance table for the identified variants
--o-table table-dada2.qza \
--o-denoising-stats dada2-stats.qza \
--verbose
```

This step will take a long time, so stop the command (ctrl+c) and copy the final output files, which have been previously made, to your current directory:

```{bash, eval=FALSE}
#Copy the files from the specified directory your working directory
cp /pub39/tea/matthew/NEOF/16s_workshop/chapter_9/* ~/Metagenetics
```

Through the above command you will now have the following three files:

- `rep-seqs-dada2.qza`: Artifact containing the identified sequence variants.
- `tabledada2.qza`: Artifact containing their ASV abundance table. 
- `dada2-stats.qza`: Artifact which summarises the statistics after the main denoising steps.

This may be one of the longest steps of the whole pipeline. The running time is proportional to the complexity of the sequences (how many ASVs are in the dataset) and to the quality of the sequences (how many errors are present). If this step is taking too long in one of your future projects, you may try to use ‘deblur’ to de-noise your data (after joining the reads with vsearch), which is generally faster.

In the case of a marker gene with very variable amplicon sequence lengths, as in the case of ITS amplicon sequencing, a possible alternative denoising strategy is to disable the filtering by truncation length using the options: `--p-trunc-len-f 0` and `--p-trunc-len-r 0`. To exclude the low quality tails the `--p-trunc-q 20` option is now required, this option will trim all the sequences after the first base with quality below ‘20’ (this may be very stringent, so you may want to try with lower values as well). A further possibility may be to use only either the forward or reverse read file, for this the `qiime dada2 denoise-single` should be use instead the above dada2 command.

## Visualise DADA2 output
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/qiime2view.png", auto_pdf = TRUE)
``` 

### table-dada2.qzv
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/table.png", auto_pdf = TRUE)
``` 
To visualise information on the ASV table we will run:

```{bash, eval=FALSE}
qiime feature-table summarize \
--i-table table-dada2.qza \
--o-visualization table-dada2.qzv \
--m-sample-metadata-file metadata.file.txt
```
 
Now load the obtained `table-dada2.qzv` into the ‘view.qiime2.org’ website.

__Tip__: Click on the big grey "Drag and drop or click here" box on the website. Then on the pop up window click "HOME" on the top right. From your home you can naigate to "Metagenetics" and choose your file.

The page opened with `table-dada2.qzv` contains 3 tabs:

1. __Overview__: An overview page with various information.
   - __Table summary__
      - __Number of samples__
      - __Number of features__: In this case features are ASVs. If this was a taxa table features would be the number of taxa.
      - __Total frequency__: Total number of sequences (i.e. total abundance across all samples and ASVs).
   - __Frequency per sample__: Summary of number of sequences.
      - __IQR table__: Table showing the Interquartile range of sample frequencies.
      - __Histogram of frequency per sample__: This gives a rough view as the start and end of the histogram is based on the minimum and maximum sample frequencies. The vertical dark blue lines on the x axis represent the specific frequencies of samples (1 vertical line per sample).
   - __Frequency per feature__: Summary of number of sequences per feature (ASV/taxa).
      - __IQR table__: Table showing the Interquartile range of feature frequencies.
      - __Histogram of frequency per sample__: This shows a rough view of the number of features (y axis) with different frequencies (x axis). Note that the scales are logarithmic. The left side of the plot shows how many rare ASVs there are (ASVs represented by a low amount of sequences, i.e. low abundance ASVs). The right side of the plot shows the amount of high abundance ASVs.
2. __Interactive Sample Detail__: This interactive section allows you to explore sampling depth. Experiment using the Plot Controls.
   - __Plot__: This shows the number of samples per metadata category.
   - __Feature count table__: Table showing feature count per sample.
   - __Plot Controls__: Controls to change and update the plot and table.
      - __Save as..__: Controls to save the current plot.
      - __Metadata Category__: This allows you to change the values on the x axis of the plot.
      - __Sampling Depth__: Choosing a sampling depth will update the following.
         - __Plot__: The plot will be updated to grey out the tops of the bars by removing samples with a depth lower than the chosen sampling depth.
         - __Feature count table__: Samples with a lower depth than the sampling depth will be highlighted red.
         - __Plot Controls__: Below the controls is a section of text showing the number of features and samples reatined by the chosen sampling depth.
3. __Feature detail__: This is a table with 3 columns.
   - 1st column: This is the ASV id. This ID represnets the exact amplicon sequence variant. This is useful as if this ASV id is in another sequencing experiment it means they are identical between the two studies. This allows for easy merging of multiple ASV tables.
   - __Frequency__: Number of sequences representing the ASV acorss all samples. I.e. the total abundance of the ASV.
   - __# of Samples Observed in__: Number of samples the ASVs was detected in.

__Questions__:

- How many ASVs were detected in the dataset?
- What is the median and mean number of sequences per samples (to 1 decimal place)?
- What is the minimum and maximum number of sequences per ASV?
- How many samples have >10,000 sequences?
- Are there more Corridor samples or Toilet samples (based on Location metadata category)?
- How many sequences represents the second most abundant ASV?

### dada2-stats.qzv
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/filter.png", auto_pdf = TRUE)
``` 
Next we'll look at the denoising stats.

```{bash, eval=FALSE}
qiime metadata tabulate \
--m-input dada2-stats.qza \
--o-visualization dada2-stats.qzv
```
 
Load the obtained ‘dada2-stats.qzv’ into the ‘view.qiime2.org’ website.

Open the `dada2-stats.qzv` visualisation with QIIME2 view.

You will now see a denoising stats table with the following columns:

- __sample-id__
- __input__: Number of sequence pairs the sample started with prior to the denoising steps.
- __filtered__: Number of sequence pairs retained after user specified filtering (e.g. `--p-trunc-len-f 220` will truncate forward reads but alos remove sequence pairs where the forward length is shorter than 220).
- __percentage of input passed filter__: Percentage of input reads that passed the filtering step.
- __denoised__: Number of filtered sequence pairs that were successfully denoised.
- __merged__: Number of filtered and denoised paired sequences that were successfully merged (forward and reverse read merged into one sequence).
- __percentage of input merged__: Percentage of input reads that passed the filtering, denoising, and merging steps.
- __non-chimeric__: Number of filtered, denoised, and merged sequences retained after chimera removal. This is the final number of sequences per sample.
- __percentage of input non-chimeric__: Percentage of input reads retained at the end.

This visualisation is very useful to see where our sequences are being filtered. It is possible to be too stringent and not hae enough sequences for some samples. This will give you an idea on how you can repeat the denoising stats with changed options to retain more samples.

__Questions__: 

- In which DADA2 step were most of the sequences lost?
- If this was a real study would there be enough sequences left for optimal analysis? 


A high percentage of sequences were detected as chimeric. In normal studies this would be worrying and may indicate a por sequencing run. However, it is fine for this tutorial as it is only practice data.