<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 Assign a taxonomic classification to each amplicon sequence variant | Bacterial 16S metabarcoding</title>
  <meta name="description" content="Chapter 10 Assign a taxonomic classification to each amplicon sequence variant | Bacterial 16S metabarcoding" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 Assign a taxonomic classification to each amplicon sequence variant | Bacterial 16S metabarcoding" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 Assign a taxonomic classification to each amplicon sequence variant | Bacterial 16S metabarcoding" />
  
  
  

<meta name="author" content="Luca Lenzi and Matthew Gemmell" />


<meta name="date" content="2021-02-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="09-ASV_identification.html"/>
<link rel="next" href="11-Phylogentic_tree.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a>Illumina QC</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Bacterial_16S_metabarcoding.html"><a href="01-Bacterial_16S_metabarcoding.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Background.html"><a href="02-Background.html"><i class="fa fa-check"></i><b>2</b> Background</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Background.html"><a href="02-Background.html#s-rrna"><i class="fa fa-check"></i><b>2.1</b> 16S rRNA</a></li>
<li class="chapter" data-level="2.2" data-path="02-Background.html"><a href="02-Background.html#its-and-other-possible-marker-genes"><i class="fa fa-check"></i><b>2.2</b> ITS and other possible marker genes</a></li>
<li class="chapter" data-level="2.3" data-path="02-Background.html"><a href="02-Background.html#amplicon-sequencing"><i class="fa fa-check"></i><b>2.3</b> Amplicon sequencing</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html"><i class="fa fa-check"></i><b>3</b> Introduction to QIIME2</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#artifacts"><i class="fa fa-check"></i><b>3.1</b> Artifacts</a></li>
<li class="chapter" data-level="3.2" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#workflow"><i class="fa fa-check"></i><b>3.2</b> Workflow</a></li>
<li class="chapter" data-level="3.3" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#prepare-the-sequence-dataset"><i class="fa fa-check"></i><b>3.3</b> Prepare the sequence dataset</a></li>
<li class="chapter" data-level="3.4" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#amplicon-sequence-variants-identification"><i class="fa fa-check"></i><b>3.4</b> Amplicon Sequence Variants identification</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#otus"><i class="fa fa-check"></i><b>3.4.1</b> OTUs</a></li>
<li class="chapter" data-level="3.4.2" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#asvs"><i class="fa fa-check"></i><b>3.4.2</b> ASVs</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#assign-a-taxonomic-classification-to-each-asv"><i class="fa fa-check"></i><b>3.5</b> Assign a taxonomic classification to each ASV</a></li>
<li class="chapter" data-level="3.6" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#build-phylogenetic-tree"><i class="fa fa-check"></i><b>3.6</b> Build phylogenetic tree</a></li>
<li class="chapter" data-level="3.7" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#data-normalisation"><i class="fa fa-check"></i><b>3.7</b> Data normalisation</a></li>
<li class="chapter" data-level="3.8" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#alpha-diversity"><i class="fa fa-check"></i><b>3.8</b> Alpha-diversity</a></li>
<li class="chapter" data-level="3.9" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#beta-diversity"><i class="fa fa-check"></i><b>3.9</b> Beta-diversity</a></li>
<li class="chapter" data-level="3.10" data-path="03-Intro_to_qiime2.html"><a href="03-Intro_to_qiime2.html#differentially-abundant-analysis"><i class="fa fa-check"></i><b>3.10</b> Differentially Abundant Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html"><i class="fa fa-check"></i><b>4</b> Cluster Introduction</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#logon-instructions"><i class="fa fa-check"></i><b>4.1</b> Logon instructions</a></li>
<li class="chapter" data-level="4.2" data-path="04-Cluster_Introduction.html"><a href="04-Cluster_Introduction.html#the-terminal-window"><i class="fa fa-check"></i><b>4.2</b> The Terminal Window</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Linux_quick_user_guide.html"><a href="05-Linux_quick_user_guide.html"><i class="fa fa-check"></i><b>5</b> Linux quick user guide</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Linux_quick_user_guide.html"><a href="05-Linux_quick_user_guide.html#accessing-the-data-and-navigating-the-command-line."><i class="fa fa-check"></i><b>5.1</b> Accessing the data and navigating the command line.</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-QIIME2_analysis_workflow.html"><a href="06-QIIME2_analysis_workflow.html"><i class="fa fa-check"></i><b>6</b> QIIME2 analysis workflow</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-QIIME2_analysis_workflow.html"><a href="06-QIIME2_analysis_workflow.html#workflow-1"><i class="fa fa-check"></i><b>6.1</b> Workflow</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="07-Import_sequences_into_QIIME2.html"><a href="07-Import_sequences_into_QIIME2.html"><i class="fa fa-check"></i><b>7</b> Import sequences into QIIME2</a></li>
<li class="chapter" data-level="8" data-path="08-Trim_PCR_sequences.html"><a href="08-Trim_PCR_sequences.html"><i class="fa fa-check"></i><b>8</b> Trim the PCR primer sequences</a></li>
<li class="chapter" data-level="9" data-path="09-ASV_identification.html"><a href="09-ASV_identification.html"><i class="fa fa-check"></i><b>9</b> De-novo amplicon sequence variants identification</a></li>
<li class="chapter" data-level="10" data-path="10-Taxonomic_assignment.html"><a href="10-Taxonomic_assignment.html"><i class="fa fa-check"></i><b>10</b> Assign a taxonomic classification to each amplicon sequence variant</a></li>
<li class="chapter" data-level="11" data-path="11-Phylogentic_tree.html"><a href="11-Phylogentic_tree.html"><i class="fa fa-check"></i><b>11</b> Make a phylogenetic tree of the identified ASVs</a></li>
<li class="chapter" data-level="12" data-path="12-Rarefaction.html"><a href="12-Rarefaction.html"><i class="fa fa-check"></i><b>12</b> Sequencing depth evaluation: rarefaction plot</a></li>
<li class="chapter" data-level="13" data-path="13-Sequencing_controls_evaluation.html"><a href="13-Sequencing_controls_evaluation.html"><i class="fa fa-check"></i><b>13</b> Sequencing controls evaluation: beta-diversity without normalisation</a></li>
<li class="chapter" data-level="14" data-path="14-Diversity_analysis.html"><a href="14-Diversity_analysis.html"><i class="fa fa-check"></i><b>14</b> Diversity analysis</a></li>
<li class="chapter" data-level="15" data-path="15-Differential_abundance.html"><a href="15-Differential_abundance.html"><i class="fa fa-check"></i><b>15</b> Differential abundance analysis</a></li>
<li class="chapter" data-level="16" data-path="16-Final_consideration.html"><a href="16-Final_consideration.html"><i class="fa fa-check"></i><b>16</b> Final consideration</a></li>
<li class="chapter" data-level="17" data-path="17-Resources.html"><a href="17-Resources.html"><i class="fa fa-check"></i><b>17</b> Resources</a></li>
<li class="chapter" data-level="18" data-path="18-Appendix.html"><a href="18-Appendix.html"><i class="fa fa-check"></i><b>18</b> Appendix</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bacterial 16S metabarcoding</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="assign-a-taxonomic-classification-to-each-amplicon-sequence-variant" class="section level1" number="10">
<h1><span class="header-section-number">Chapter 10</span> Assign a taxonomic classification to each amplicon sequence variant</h1>
<p><img src="figures/classification.png" width="20%" style="display: block; margin: auto;" /></p>
<p>Now that we have defined the sequence variants in the samples, we can assign taxonomic labels to them. Here, we use the <code>sk-learn</code> classifier to assign taxonomy based on the Greengenes database (DeSantis <em>et al</em>., 2006).</p>
<p>Greengenes is a database containing quality-controlled microbial sequence data. Greengenes 16S rDNA sequences come from a large number of species which have been formatted for use in QIIME, by clustering the sequences at a given level of similarity (here, 99%). As for OTU-picking, the more stringent the similarity threshold, the more fine-grained the taxonomic designation (i.e. there are more Greengenes OTUs defined to the species level at 99% than at 97% similarity). The classifier maps the variant sequences, defined above, to OTU representative sequences from the Greengenes set.</p>
<p>First, we need to import the Greengenes data into QIIME2 artifact objects.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="10-Taxonomic_assignment.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools import <span class="kw">\</span></span>
<span id="cb23-2"><a href="10-Taxonomic_assignment.html#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="ex">--input-path</span></span>
<span id="cb23-3"><a href="10-Taxonomic_assignment.html#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="ex">/pub39/tea/luca/Metagenetics_2019/db/gg_13_8_otus/rep_set/99_otus.fasta</span> <span class="kw">\</span></span>
<span id="cb23-4"><a href="10-Taxonomic_assignment.html#cb23-4" aria-hidden="true" tabindex="-1"></a> <span class="ex">--output-path</span> gg-13-8.99_otus.qza <span class="kw">\</span></span>
<span id="cb23-5"><a href="10-Taxonomic_assignment.html#cb23-5" aria-hidden="true" tabindex="-1"></a> <span class="ex">--type</span> <span class="st">&#39;FeatureData[Sequence]&#39;</span></span>
<span id="cb23-6"><a href="10-Taxonomic_assignment.html#cb23-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-7"><a href="10-Taxonomic_assignment.html#cb23-7" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> tools import <span class="kw">\</span></span>
<span id="cb23-8"><a href="10-Taxonomic_assignment.html#cb23-8" aria-hidden="true" tabindex="-1"></a><span class="ex">--input-path</span></span>
<span id="cb23-9"><a href="10-Taxonomic_assignment.html#cb23-9" aria-hidden="true" tabindex="-1"></a><span class="ex">/pub39/tea/luca/Metagenetics_2019/db/gg_13_8_otus/taxonomy/99_otu_taxonomy.txt</span> <span class="kw">\</span></span>
<span id="cb23-10"><a href="10-Taxonomic_assignment.html#cb23-10" aria-hidden="true" tabindex="-1"></a> <span class="ex">--output-path</span> gg-13-8.99.taxa.qza <span class="kw">\</span></span>
<span id="cb23-11"><a href="10-Taxonomic_assignment.html#cb23-11" aria-hidden="true" tabindex="-1"></a> <span class="ex">--input-format</span> HeaderlessTSVTaxonomyFormat <span class="kw">\</span></span>
<span id="cb23-12"><a href="10-Taxonomic_assignment.html#cb23-12" aria-hidden="true" tabindex="-1"></a> <span class="ex">--type</span> <span class="st">&#39;FeatureData[Taxonomy]&#39;</span></span></code></pre></div>
<p>These steps will create two objects containing the sequences and their taxonomy, respectively, and should finish in less than 2 mins.</p>
<p>The classifier tool we are going to use requires a training step to reach its optimum performance. The training is performed on a read set extracted from Greengenes representative sequences, including only the amplified region of interest.</p>
<p>Note: Do not run the below command for this tutorial, see below command)</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb24-1"><a href="10-Taxonomic_assignment.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> feature-classifier extract-reads <span class="kw">\</span></span>
<span id="cb24-2"><a href="10-Taxonomic_assignment.html#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="ex">--i-sequences</span> gg-13-8.99_otus.qza <span class="kw">\</span></span>
<span id="cb24-3"><a href="10-Taxonomic_assignment.html#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="ex">--p-f-primer</span> NNNNNGTGCCAGCMGCCGCGGTAA <span class="kw">\</span></span>
<span id="cb24-4"><a href="10-Taxonomic_assignment.html#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="ex">--p-r-primer</span> GGACTACHVGGGTWTCTAAT <span class="kw">\</span></span>
<span id="cb24-5"><a href="10-Taxonomic_assignment.html#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating reads of 250 bp as in our case</span></span>
<span id="cb24-6"><a href="10-Taxonomic_assignment.html#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="ex">--p-trunc-len</span> 250 <span class="kw">\</span></span>
<span id="cb24-7"><a href="10-Taxonomic_assignment.html#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="ex">--o-reads</span> gg-13-8.99.ref.seqs.qza</span></code></pre></div>
<p>Given that this step may require more than 14h to finish, we won’t be able to run on this occasion. Copy over premade result with the following command:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb25-1"><a href="10-Taxonomic_assignment.html#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> /pub39/tea/luca/Metagenetics_2019/gg-13-8.99.ref.seqs.qza .</span></code></pre></div>
<p>To train the <code>sk-learn</code> classifier on this set of reads (approx. 25 mins):</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb26-1"><a href="10-Taxonomic_assignment.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> feature-classifier fit-classifier-naive-bayes <span class="kw">\</span></span>
<span id="cb26-2"><a href="10-Taxonomic_assignment.html#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="ex">--i-reference-reads</span> gg-13-8.99.ref.seqs.qza <span class="kw">\</span></span>
<span id="cb26-3"><a href="10-Taxonomic_assignment.html#cb26-3" aria-hidden="true" tabindex="-1"></a> <span class="ex">--i-reference-taxonomy</span> gg-13-8.99.taxa.qza <span class="kw">\</span></span>
<span id="cb26-4"><a href="10-Taxonomic_assignment.html#cb26-4" aria-hidden="true" tabindex="-1"></a> <span class="ex">--o-classifier</span> classifier.trained.qza</span></code></pre></div>
<p>The command for the taxonomy assignment is (approx. 5 min):</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb27-1"><a href="10-Taxonomic_assignment.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> feature-classifier classify-sklearn <span class="kw">\</span></span>
<span id="cb27-2"><a href="10-Taxonomic_assignment.html#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="ex">--i-classifier</span> classifier.trained.qza <span class="kw">\</span></span>
<span id="cb27-3"><a href="10-Taxonomic_assignment.html#cb27-3" aria-hidden="true" tabindex="-1"></a> <span class="co"># Default value for confidence</span></span>
<span id="cb27-4"><a href="10-Taxonomic_assignment.html#cb27-4" aria-hidden="true" tabindex="-1"></a> <span class="ex">--p-confidence</span> 0.7 <span class="kw">\</span></span>
<span id="cb27-5"><a href="10-Taxonomic_assignment.html#cb27-5" aria-hidden="true" tabindex="-1"></a> <span class="ex">--i-reads</span> rep-seqs-dada2.qza <span class="kw">\</span></span>
<span id="cb27-6"><a href="10-Taxonomic_assignment.html#cb27-6" aria-hidden="true" tabindex="-1"></a> <span class="ex">--o-classification</span> taxonomy.sklearn.qza</span></code></pre></div>
<p>To visualise the taxonomy-abundance bar-chart obtained, run (approx. 1 min):</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb28-1"><a href="10-Taxonomic_assignment.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="ex">qiime</span> taxa barplot --i-table table-dada2.qza <span class="kw">\</span></span>
<span id="cb28-2"><a href="10-Taxonomic_assignment.html#cb28-2" aria-hidden="true" tabindex="-1"></a> <span class="ex">--i-taxonomy</span> taxonomy.sklearn.qza <span class="kw">\</span></span>
<span id="cb28-3"><a href="10-Taxonomic_assignment.html#cb28-3" aria-hidden="true" tabindex="-1"></a> <span class="ex">--m-metadata-file</span> metadata.file.txt <span class="kw">\</span></span>
<span id="cb28-4"><a href="10-Taxonomic_assignment.html#cb28-4" aria-hidden="true" tabindex="-1"></a> <span class="ex">--o-visualization</span> taxa-bar-plots.dada2.qzv</span></code></pre></div>
<p>Once you have loaded the file in the browser, you may specify the taxonomic level to look at, as well as a few sorting and colouring options.</p>
<p><strong>Questions</strong>:</p>
<ul>
<li>What is the most abundant taxon, at level 5, in the dataset?</li>
<li>How do the samples compare to each other?</li>
</ul>
<p>If you have sequenced any samples with known composition (e.g. mock community), now is the time to see if they behave as expected, you may want to change some step and/or settings of what you have done so far if the composition does not match what you expect. For example, change the de-noising tool or apply more quality filters to further improve the quality of the sequence under examination.</p>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="09-ASV_identification.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="11-Phylogentic_tree.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
