--- 
title: "Bacterial 16S metabarcoding"
author: "Luca Lenzi and Matthew Gemmell"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
---

```{r, fig.align = 'center',out.width= '30%', echo=FALSE }
knitr::include_graphics(path = "figures/NEOF.png", auto_pdf = TRUE)
``` 

# Introduction
```{r, fig.align = 'center',out.width= '40%', echo=FALSE }
knitr::include_graphics(path = "figures/intro_logo.png", auto_pdf = TRUE)
``` 

This practical session aims to introduce you to the analysis of bacterial 16S metabarcoding with QIIME2. The topics covered are:

- Background on the biology
- Introduction to QIIME2
- Cluster and webVNC information
- Information on initial data
- QIIME2 analysis workflow
- Sequence import
- PCR primer trimming
- DADA2 denoising
- Taxonomic classification
- Phylogenetic tree construction
- Rarefaction curve
- Alpha and beta diversity analysis
- Differential abundance analysis
- Final considerations