---
title: "Bacterial 16S metabarcoding"
author: "Luca Lenzi and Matthew R. Gemmell"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
favicon: figures/NEOF_favicon.png
description: NEOF book for the Bacterial 16S metabarcoding workshop
cover-image: "figures/NEOF.png"
---

```{r cite-packages, include = FALSE}
# automatically create a bib database for R packages
# add any packages you want to cite here
knitr::write_bib(c(
  .packages(), 'bookdown', 'webexercises'
), 'packages.bib')
```

```{r, echo=FALSE}
#Change colour, border, and text of code chunks
#Check style.css for .Rchunk
#https://stackoverflow.com/questions/65627531/change-r-chunk-background-color-in-bookdown-gitbook
#https://bookdown.org/yihui/rmarkdown-cookbook/chunk-styling.html
knitr::opts_chunk$set(class.source="Rchunk") 
```

```{r, fig.align = 'center',out.width= '30%', echo=FALSE }
knitr::include_graphics(path = "figures/NEOF.png", auto_pdf = TRUE)
```

# Introduction

```{r, fig.align = 'center',out.width= '30%', echo=FALSE }
knitr::include_graphics(path = "figures/intro_logo.png", auto_pdf = TRUE)
```

This practical session aims to introduce you to the analysis of bacterial 16S metabarcoding with QIIME2. The topics covered are:

```{r, echo=FALSE, align="centre"}
#Create 2 coilumn dataframe
column_1 <- c(
 '[**Background on the biology**](#background)', 
 '[![](figures/background.png){height="150"}](#background)',
 '[**Cluster and webVNC information**](#cluster)',
 '[![](figures/cluster.png){height="150"}](#cluster)',
 '[**QIIME2 analysis workflow**](#q2workflow)',
 '[![](figures/workflow.png){height="150"}](#q2workflow)',
 '[**PCR primer trimming**](#pcrtrim)',
 '[![](figures/prepare.png){height="150"}](#pcrtrim)',
 '[**Taxonomic classification**](#taxa)',
 '[![](figures/classification.png){height="150"}](#taxa)',
 '[**Rarefaction curve**](#rarefaction)',
 '[![](figures/rarefaction.png){height="150"}](#rarefaction)',
 '[**Differential abundance analysis**](#differential)',
 '[![](figures/balance_scale.png){height="150"}](#differential)',
 '[**Appendix**](#resources)',
 '[![](figures/resources.png){height="150"}](#resources)'
)
column_2 <- c(
  '[**Introduction to QIIME2**](#introqiime2)',
 '[![](figures/intro_logo.png){height="150"}](#introqiime2)',
 '[**Introduction to QIIME2**](#introqiime2)',
 '[![](figures/data_files.png){height="150"}](#data)',
 '[**Sequence import**](#import)',
 '[![](figures/import.png){height="150"}](#import)',
 '[**DADA2 denoising**](#dada2)',
 '[![](figures/clusters.png){height="150"}](#dada2)',
 '[**Phylogenetic tree construction**](#phylogeny)',
 '[![](figures/phylo_tree.png){height="150"}](#phylogeny)',
 '[**Alpha and beta diversity analysis**](#diversity)',
 '[![](figures/diversity.png){height="150"}](#diversity)',
 '[**Final considerations**](#final)',
 '[![](figures/finish.png){height="150"}](#final)',
 "",""
)
table_df <- data.frame(column_1, column_2)
#Kable and kableextra libraries
library(knitr)
library(kableExtra)
#Create table
#ALign both column in centre (align = "cc")
table_df %>%
  kbl(align= "cc", col.names = NULL) %>%
  kable_styling(position = "center")
```

<a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/4.0/"><img src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" alt="Creative Commons Licence" style="border-width:0"/></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/4.0/">Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License</a>.
