--- 
title: "Bacterial 16S metabarcoding"
author: "Luca Lenzi and Matthew Gemmell"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
favicon: figures/NEOF_favicon.png
description: NEOF book for the Bacterial 16S metabarcoding workshop
cover-image: "figures/NEOF.png"
---

```{r cite-packages, include = FALSE}
# automatically create a bib database for R packages
# add any packages you want to cite here
knitr::write_bib(c(
  .packages(), 'bookdown', 'webexercises'
), 'packages.bib')
```

```{r, fig.align = 'center',out.width= '30%', echo=FALSE }
knitr::include_graphics(path = "figures/NEOF.png", auto_pdf = TRUE)
``` 

# Introduction
```{r, fig.align = 'center',out.width= '30%', echo=FALSE }
knitr::include_graphics(path = "figures/intro_logo.png", auto_pdf = TRUE)
``` 

This practical session aims to introduce you to the analysis of bacterial 16S metabarcoding with QIIME2. The topics covered are:

- [Background on the biology](#background)
- [Introduction to QIIME2](#introqiime2)
- [Cluster and webVNC information](#cluster)
- [Information on initial data](#data)
- [QIIME2 analysis workflow](#q2workflow)
- [Sequence import](#import)
- [PCR primer trimming](#pcrtrim)
- [DADA2 denoising](#dada2)
- [Taxonomic classification](#taxa)
- [Phylogenetic tree construction](#phylogeny)
- [Rarefaction curve](#rarefaction)
- [Alpha and beta diversity analysis](#diversity)
- [Differential abundance analysis](#differential)
- [Final considerations](#final)

<a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/4.0/"><img alt="Creative Commons Licence" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/4.0/">Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License</a>.