# QIIME2 analysis workflow {#q2workflow}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/workflow.png", auto_pdf = TRUE)
``` 

## Workflow

The 'QIIME2' analysis workflow comprises of nine steps:

1. Import sequences into QIIME2 artifact
2. Trim the PCR primer sequences
3. De-novo amplicon sequence variants identifications
4. Assign taxonomic classification to each ASV
5. Create a phylogenetic tree of the identified ASVs
6. Sequencing depth evaluation: rarefaction plot
7. Diversity analysis: alpha and beta
8. Differential abundance analysis of ASVs across different conditions

## Activate environment
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/mamba_logo.png", auto_pdf = TRUE)
``` 
You will need to activate the QIIME2 `mamba` environment before continuing. Carry this out with the following command.

__Note__: Ensure you include the dot and space (`. `) at the start before `useqiime2-2023.5`

```{bash eval=FALSE}
. useqiime2-2023.5
```