# QIIME2 analysis workflow {#q2workflow}
<center>
![](figures/workflow.png){style="width:200px; background: white; border-radius:5px"}
</center>

## Workflow

The 'QIIME2' analysis workflow comprises of nine steps:

1. Import sequences into QIIME2 artifact
2. Trim the PCR primer sequences
3. De-novo amplicon sequence variants identifications
4. Assign taxonomic classification to each <u id='asv_tip'>ASV</u>
5. Create a phylogenetic tree of the identified <u id='asv_tip'>ASVs</u>
6. Sequencing depth evaluation: rarefaction plot
7. Diversity analysis: <u id='alpha_tip'>alpha</u> and <u id='beta_tip'>beta</u>
8. Differential abundance analysis of <u id='asv_tip'>ASVs</u> across different conditions

## Activate environment
<center>
![](figures/mamba_logo.png){style="width:200px; border-radius:5px"}
</center>

You will need to activate the QIIME2 `mamba` environment before continuing. Carry this out with the following command.

__Note__: Ensure you include the dot and space (`. `) at the start before `useqiime2-amplicon-2024.10`

```{bash eval=FALSE}
. useqiime2-amplicon-2024.10
```

Information on installing `mamba` and creating the environment are in the [appendix](#mamba_install)

```{r, echo=FALSE}
#Tippy tooltips
tippy::tippy_this(elementId = "asv_tip", 
                  tooltip = "Amplicon Sequence Variant",
                  arrow = TRUE, placement = "bottom")
tippy::tippy_this(elementId = "alpha_tip", 
                  tooltip = "Alpha diversity values are based on single samples",
                  arrow = TRUE, placement = "bottom")
tippy::tippy_this(elementId = "beta_tip", 
                  tooltip = "Beta diversity values based on pairwise comparisons",
                  arrow = TRUE, placement = "bottom")
```