# (PART\*) Preprocessing of data {-}

# Sequence import {#import}
<center>
![](figures/import.png){style="width:200px; background: white; border-radius:5px"}
</center>

Before we can work with our data in QIIME2 we need to import the data into a QIIME2 <u id='artifact_tip'>artifact</u>.

## Import: run
<center>
![](figures/play_green.png){style="width:150px; background: white; border: white solid 5px; border-radius:5px"}
</center>

Import our read pairs into a QIIME2 <u id='artifact_tip'>artifact</u>.

__Note__: It is quite easy to make typos in the following command. Ensure to read any errors carefully to try to figure out where a typo is, if any.

__Note 2__: If you are not familiar with the back slashes (`\`) in the below command please check our explanation of __[Bash escape](https://neof-workshops.github.io/Unix_nxcdf7/Course/05-Tips_and_tricks.html#bash-escape)__. Please also ask for a demonstration if you would like or you can watch a [video demonstration on YouTube](https://www.youtube.com/watch?v=968QURuAplk&t=607s).

```{bash, eval=FALSE}
qiime tools import \
--type 'SampleData[PairedEndSequencesWithQuality]' \
--input-path PairedEndFastqManifestPhred33.csv \
--output-path paired-end-demux.qza \
--input-format PairedEndFastqManifestPhred33
```

#### Parameters {-}
<center>
![](figures/parameter_green.png){style="width:100px; background: white; border: white solid 5px; border-radius:5px"}
</center>

- __`--type`__: The type of data being imported.
  - `'SampleData[PairedEndSequencesWithQuality]'` specifies we are importing Paired end sequence data that contains quality information. I.e. normal Illumina paired end fastq data.
- __`--input-path`__: The input [manifest file](#manifestfile).
- __`--output-path`__: Output QIIME2 <u id='artifact_tip'>artifact</u> file containing your sequence data.
  - The "demux" part of the file name indicates that the samples are already demultiplexed (i.e. sequences have been separated by samples). 
  - QIIME2 <u id='artifact_tip'>artifacts</u> use "-" as word separators in file names.  This is not mandatory but it is a good practice to keep consistent with the QIIME2 developers.
- __`--input-format`__: The format of the input files.
  - `PairedEndFastqManifestPhred33`: Our input file is a [Manifest file](#manifestfile) with information on paired fastq files. Our fastq files have quality encoding of Phred33, the most commonly used for Illumina data now. More information about fastq quality encoding can be found on [wikipedia](#https://en.wikipedia.org/wiki/FASTQ_format#Encoding).
 
[QIIME2 docs on importing data](https://docs.qiime2.org/2024.10/tutorials/importing/)

## Import: visualise {#chap7_vis}
<center>
![](figures/eye_green.png){style="width:200px; background: white; border: white solid 5px; border-radius:5px"}
</center>

To visualise the samples loaded, the following command will create a visualisation <u id='artifact_tip'>artifact</u>.

```{bash, eval=FALSE}
qiime demux summarize \
--i-data paired-end-demux.qza \
--o-visualization paired-end-demux.qzv
```

#### Parameters {-}
<center>
![](figures/parameter_green.png){style="width:100px; background: white; border: white solid 5px; border-radius:5px"}
</center>

- __`--i-data`__: Input data <u id='artifact_tip'>artifact</u> (`.qza`) to be summarised and visualised.
- __`--o-visualization`__: Output visualisation <u id='artifact_tip'>artifact</u> (`qzv`).

[QIIME2 docs on `qiime demux summarize`](https://docs.qiime2.org/2024.10/plugins/available/demux/summarize/)

#### Firefox {-}
<center>
![](figures/firefox.png){style="width:100px; background: white; border: white solid 5px; border-radius:5px"}
</center>

We have created our visualistion file, now we need to visualise it. We carry this out with the website [view.qiime2.org](https://view.qiime2.org).

First we need to open up `Firefox`. Right click on the main screen (greyish background) and choose `firefox` from the dropdown menu.

Then:

- Navigate to the following web-site on `firefox`: ‘https://view.qiime2.org’.
- Click on the ‘Drag and drop or click here’ grey box.
- Use the file browser to find and select the file ‘paired-end-demux.qzv’. 

__Tip__: click "Home" on the top left, from there you can navigate to "Metagenetics" and choose your file.

#### Output {-#chap7output}
<center>
![](figures/output_file_green.png){style="width:100px"}
</center>

The visualisation will have two tabs:

1. __Overview__: This contains 3 main sections relating to the sequence count of samples.
   - __Demultiplexxed sequence counts summary__: Summary table of number of sequences per sample divided by forward and reverse reads.
   - __Read frequency histograms__: Forward and Reverse reads histograms showing how many samples have various numbers of sequences. 
   - __Per-sample sequence counts__: Table showing the number of forward and reverse reads in samples.
2. __Interactive Quality Plots__: Contains similar information as fastQC analysis.
   - __Interactive quality plots__: These show the quality of your forward and reverse reads using a random sampling of 10000 reads. You can click on a part of the plot to get info in the below "Parametric seven-number summary" table.
   - __Demultiplexed sequence length summary__: Table showing summary of sequence lengths for the forward and reverse reads.

## Import: MCQs
<center>
![](figures/question_bubble_green.png){style="width:150px; background: white; border: white solid 5px; border-radius:5px"}
</center>

```{r, echo = FALSE}
opts_p <- c(answer="__42,853__", "__126,670__", "__174,951__")
```
1. What is the minimum number of forward reads across the samples? `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c("__42,853__", "__126,670__", answer="__174,951__")
```
2. What is the maximum number of forward reads across the samples? `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c("__42,853__", answer="__126,670__", "__174,951__")
```
3. What is the mean number of forward reads across the samples? `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c("__1TM__", "__3K2E__", answer="__3K1M__")
```
4. Which sample has 134,737 forward reads? `r longmcq(opts_p)`
 
```{r, echo = FALSE}
opts_p <- c(answer="__1TM__", "__3K2E__", "__3K1M__")
```
5. Which sample has 86,083 forward reads? `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c(answer="__TRUE__", "__FALSE__")
```
6. Is most of the read quality for the forward reads above Q30? `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c(answer="__TRUE__", "__FALSE__")
```
7. Is most of the read quality for the reverse reads above Q30? `r longmcq(opts_p)`

## Import: summary
<center>
![](figures/sum_green.png){style="width:150px; background: white; border: white solid 5px; border-radius:5px"}
</center>

Brilliant! We have imported our read data and checked the number of reads per sample which looks good. The next step is to remove data from the sequence data we do not need any more.

```{r, echo=FALSE}
#Tippy tooltips
tippy::tippy_this(elementId = "s_tip", 
                  tooltip = "16S ribosoma RNA",
                  arrow = TRUE, placement = "bottom")
tippy::tippy_this(elementId = "otu_tip", 
                  tooltip = "Operational Taxonomy Unit",
                  arrow = TRUE, placement = "bottom")
tippy::tippy_this(elementId = "asv_tip", 
                  tooltip = "Amplicon Sequence Variant",
                  arrow = TRUE, placement = "bottom")
tippy::tippy_this(elementId = "alpha_tip", 
                  tooltip = "Alpha diversity values are based on single samples",
                  arrow = TRUE, placement = "bottom")
tippy::tippy_this(elementId = "beta_tip", 
                  tooltip = "Beta diversity values based on pairwise comparisons",
                  arrow = TRUE, placement = "bottom")
tippy::tippy_this(elementId = "artifact_tip", 
                  tooltip = "Artifact files are specific to QIIME2 and can contain data (.qza) or visualisations (.qzv)",
                  arrow = TRUE, placement = "bottom")
tippy::tippy_this(elementId = "amplicon_tip", 
                  tooltip = "Sequences created by amplifying (e.g. PCR) and sequencing (e.g. Illumina) a targeted region of DNA (e.g. 16S rRNA region)",
                  arrow = TRUE, placement = "bottom")
```