# Phylogenetic tree construction {#phylogeny}
<center>
![](figures/phylo_tree.png){style="width:200px; background: white; border-radius:5px"}
</center>

In this section we will obtain a phylogenetic tree of our <u id='asv_tip'>ASV</u> representative sequences. 
The final rooted tree will be used to compute phylogenetically aware <u id='alpha_tip'>alpha-</u> and <u id='beta_tip'>beta-</u>diversity metrics such as Faith’s Phylogenetic Distance (Faith’s PD) and UniFrac distances. 
This is only carried out on marker regions for which this is possible (therefore excluding ITS and others).

## MAFFT alignment
<center>
![](figures/dna_alignment.png){style="width:200px; background: white; border-radius:5px"}
</center>

The first step is to align our representative sequences against each other. 
This is very straightforward as <u id='s_tip'>16S</u> amplicon sequences are relatively similar and short in length.

```{bash, eval=FALSE}
qiime alignment mafft \
--i-sequences rep-seqs-dada2.qza \
--o-alignment aligned-rep-seqs.qza
```

#### Parameters{-}
<center>
![](figures/parameter_red.png){style="width:100px; border-radius: 5px ; background: white; border: 5px solid white"}
</center>

- __`--i-sequences`__: Input <u id='artifact_tip'>artifact</u> containing the representative sequences of the features (<u id='asv_tip'>ASVs</u>) to be aligned.
- __`--o-alignment`__: Output <u id='artifact_tip'>artifact</u> containing the aligned representative sequences.

[QIIME2 docs on `qiime alignment mafft`](https://amplicon-docs.qiime2.org/en/latest/references/plugins/alignment.html#q2-action-alignment-mafft)

## Masking
<center>
![](figures/mask.png){style="width:150px; border-radius: 5px ; background: white"}
</center>

After alignment non-informative regions are masked. This will make our phylogenetic construction quicker and better.

```{bash, eval=FALSE}
qiime alignment mask \
--i-alignment aligned-rep-seqs.qza \
--p-max-gap-frequency 1.0 \
--p-min-conservation 0.4 \
--o-masked-alignment masked-aligned-rep-seqs.qza
```

#### Parameters{-}
<center>
![](figures/parameter_blue.png){style="width:100px; border-radius: 5px ; background: white; border: 5px solid white"}
</center>

- __`--i-alignment`__: Input <u id='artifact_tip'>artifact</u> containing aligned representative sequences.
- __`--p-max-gap-frequency`__: This determines the relative frequency of gap characters in an aligned column for the column to be retained.
  - This option is provided with a number from 0-1.
  - 0.0 = Columns consisting of <=0% gaps will be retained. I.e. a column must have no gaps to be retained.
  - 1.0 = Columns consisting of <=100% gaps will be retained. I.e. no column will be removed based on number of gaps.
  - 1.0 was chosen as it is the default option. Unless you have good reason this is what you should use for your future 16S analysis.
- __`--p-min-conservation`__: The minimum relative frequency of at least one non-gap character (e.g. A,G,C,T) in a column for that column to be retained.
  - This option is provided with a number from 0-1.
  - 0.4 = A column will  be retained if it contains at least one character that is present in at least 40% of the sequences. 
    - E.g. a column consisting of 25% A, 25% G, 25% C, and 25% T will __not__ be retained.
    - A column consisting of 40% A, 20% G, 20% C, and 20% T __will__ be retained.
    - 0.4 was chosen as it is the default. Unless you have good reason this is what you should use for your future 16S analysis.
- __`--o-masked-alignment`__: Output <u id='artifact_tip'>artifact</u> containing the masked aligned representative sequences.

[QIIME2 docs on `qiime alignment mask`](https://amplicon-docs.qiime2.org/en/latest/references/plugins/alignment.html#q2-action-alignment-mask)

## Un-rooted tree creation with FASTTREE
<center>
![](figures/tree_unrooted.png){style="width:200px; border-radius: 5px ; background: white; border: 5px solid white"}
</center>

Create an un-rooted maximum-likelihood tree from our masked alignment with the Fasttree 2 tool. 

```{bash, eval=FALSE}
qiime phylogeny fasttree \
--i-alignment masked-aligned-rep-seqs.qza \
--o-tree unrooted-tree.qza
```

#### Parameters{-}
<center>
![](figures/parameter_purple.png){style="width:100px; border-radius: 5px ; background: white; border: 5px solid white"}
</center>

- __`--i-alignment`__: Input artifact containing the masked aligned representative sequences.
- __`--o-tree`__: Output artifact containing the un-rooted tree.

[QIIME2 docs on `qiime phylogeny fasttree`](https://amplicon-docs.qiime2.org/en/latest/references/plugins/phylogeny.html#q2-action-phylogeny-fasttree)

## Rooted tree creation
<center>
![](figures/tree_rooted.png){style="width:150px; border-radius: 5px ; background: white"}
</center>

Finally the un-rooted tree is rooted.

```{bash, eval=FALSE}
qiime phylogeny midpoint-root \
--i-tree unrooted-tree.qza \
--o-rooted-tree rooted-tree.qza
```

#### Parameters{-}
<center>
![](figures/parameter_black.png){style="width:100px; border-radius: 5px ; background: white; border: 5px solid white"}
</center>

- __`--i-tree`__: Input <u id='artifact_tip'>artifact</u> containing the un-rooted tree
- __`--o-rooted-tree`__: Output <u id='artifact_tip'>artifact</u> containing the rooted tree.

[QIIME2 docs on `qiime phylogeny midpoint-root`](https://amplicon-docs.qiime2.org/en/latest/references/plugins/phylogeny.html#q2-action-phylogeny-midpoint-root)

## Phylogeny: summary
<center>
![](figures/sum_black.png){style="width:100px; border-radius: 5px ; background: white; border: 5px solid white"}
</center>

All these steps should finish in approximately 3 minutes. The final <u id='artifact_tip'>artifact</u> for later use is ‘rooted-tree.qza’. This is a rooted tree that will be used for some biodiversity measures that require phylogenetic information.

The file ‘rooted-tree.qza’ contains a phylogenetic tree that can be exported to a newick file. This is covered in the [supplemental book](https://neof-workshops.github.io/16S_yg84o9/Supplemental/02-Exporting_artifacts.html#phylogenetic-tree-export).

```{r, echo=FALSE}
#Tippy tooltips
tippy::tippy_this(elementId = "s_tip", 
                  tooltip = "16S ribosoma RNA",
                  arrow = TRUE, placement = "bottom")
tippy::tippy_this(elementId = "otu_tip", 
                  tooltip = "Operational Taxonomy Unit",
                  arrow = TRUE, placement = "bottom")
tippy::tippy_this(elementId = "asv_tip", 
                  tooltip = "Amplicon Sequence Variant",
                  arrow = TRUE, placement = "bottom")
tippy::tippy_this(elementId = "alpha_tip", 
                  tooltip = "Alpha diversity values are based on single samples",
                  arrow = TRUE, placement = "bottom")
tippy::tippy_this(elementId = "beta_tip", 
                  tooltip = "Beta diversity values based on pairwise comparisons",
                  arrow = TRUE, placement = "bottom")
tippy::tippy_this(elementId = "artifact_tip", 
                  tooltip = "Artifact files are specific to QIIME2 and can contain data (.qza) or visualisations (.qzv)",
                  arrow = TRUE, placement = "bottom")
tippy::tippy_this(elementId = "amplicon_tip", 
                  tooltip = "Sequences created by amplifying (e.g. PCR) and sequencing (e.g. Illumina) a targeted region of DNA (e.g. 16S rRNA region)",
                  arrow = TRUE, placement = "bottom")
```