#  Sequencing depth evaluation: rarefaction plot
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/rarefaction.png", auto_pdf = TRUE)
``` 

The detected diversity partly depends on the sampling depths. An incompletely sampled community will appear less diverse than a fully sampled community. 

To compare the diversity between samples we need to account for their different sampling depths. This can be carried out with rarefaction. 
This is performed by reducing all the samples at the same sequence count, by random sub-sampling of the sequences.

For example, if we rarefy to a depth of 2000 for all samples:

- Samples with depths of <2000 will be removed.
- Samples with depths of 2000 will be unchanged.
- Samples with depths of >2000 will have all reads removed except for a random subsample of 2000.

Equally sized subsamples are directly comparable among different samples. Additionally, we can progressively sample smaller subsamples from a sample to draw a curve of estimated alpha diversity. The shape of this curve is informative. If the curve plateaus, then increased sampling depth will not increase the estimate of alpha diversity (the community is fully sampled). 

```{r, fig.align = 'center',out.width= '80%', echo=FALSE }
knitr::include_graphics(path = "figures/rarefaction_example.png", auto_pdf = TRUE)
``` 
Example rarefaction curve: We can see that Series 1 plateaus at ~7,000 sequences and Series 3 plateaus at ~18,000 sequences whilst Series 2 has not started to plateau.

Given an abundance table (and a phylogenetic tree for calculating phylogenetic distance), we can plot the rarefaction curve using different alpha-diversity metrics. Here, we will use diversity metrics (observed features, Simpson, Shannon, Faith’s PD) as well as an evenness metric (Simpson’s evenness).

__Note__: See the appendix at the end of this bookdown for details on each metric.

You may want to exclude `--i-phylogeny rooted-tree.qza` and `--p-metrics faith_pd` for markers such as ITS where you will not create a phylogenetic tree.

```{bash, eval=FALSE}
qiime diversity alpha-rarefaction --i-table table-dada2.qza \
 --i-phylogeny rooted-tree.qza \
 # Min and max depth for the plot
 --p-min-depth 5000 \
 --p-max-depth 130000 \
 # Evaluating diversity every 75 seqs depth
 --p-steps 75 \
 # Using average from 55 iterations for the diversity count
 --p-iterations 55 \
 --m-metadata-file metadata.file.txt \
 --p-metrics simpson_e \
 --p-metrics simpson \
 --p-metrics shannon \
 --p-metrics observed_features \
 --p-metrics faith_pd \
 --o-visualization rarefaction-curve.qzv
```

The command will take approximately 10 minutes.

The result is a visualisation artifact. Load the file into QIIME2 view. You can select the metric to visualise as well as the grouping (there is one non-numerical column in the metadata file).

The visualisation will have two plots. 

- Top plot = Alpha rarefaction plot.
   - Primarily used to determine if the richness of the samples has been fully observed or sequenced. 
   - If the lines in the plot appear to “level out” (i.e., approach a slope of zero) at some sampling depth along the xaxis, this suggests that collecting additional sequences beyond that sampling depth would not likely result in the observation of additional features. 
   - If the lines in a plot don’t level out:
      - The richness of the samples  may have not been fully observed (because too few sequences were collected)
      - Or it could be an indicator that a lot of sequencing errors remain in the data (which is being mistaken for novel diversity).
- Bottom plot.
   - This is important when grouping samples by metadata.
   - It illustrates the number of samples that remain in each group through the increasing of sampling depth.
   - This is important as when samples are removed it will change the rarefaction value of the group. This is because that sample is no longer considered for the calculation. This can be seen by jitters in the rarefaction curve corresponding a decrease in number of samples.

Choosing a rarefaction threshold for the normalisation step, is ultimately identifying the best tradeoff between:

- Losing sequencing depth within samples.
- Losing entire samples.

The best choice usually depends on the experimental design. If you have many samples per group you may be able to lose a few samples without losing statistical power for the analysis. If you have few samples per group you may need to choose a lower threshold.

Other considerations for rarefaction threshold:

- Does the samples in one group have lower depths? Although you may be able to retain all the samples in the other groups with a high rarefaction threshold you may remove too much of the low depth group.
- Should you samples have relatively high (e.g. soil microbiomes) or low (e.g. human gut microbiome) diveristy. With higher diveristy you will need a larger rarefaction threshold

__Questions__: 

- Are any samples not fully sequenced? 
- Choose a sampling depth (rarefaction threshold) to use for the alpha- and beta-diversity analysis.