# Sequencing controls evaluation: beta-diversity without normalisation
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/control.png", auto_pdf = TRUE)
``` 

Current protocols for metagenetic experiments strongly suggest to always include positive and negative controls. It is therefore likely that in your design you will have:

- Negative controls to explore the microbiome derived by the kits used, the so-called ‘kit-ome’ 
- Positive controls (mock or known communities) to investigate biases from the lab protocols or the bioinformatics pipeline).

To investigate if the controls (especially the negatives) show contaminations and or suspicious similarity with the samples under investigation, it is a good practice to investigate beta-diversities before proceeding with the rarefaction step. This is because the rarefaction will probably exclude most of the negative controls as they generally consist of a low number of sequences.

In the experimental design used for this practical there are no controls. However, it may be good to look at the diversity before rarefaction to compare with the same after rarefaction. The following commands are used to obtain the Principal Coordinate Analysis (PCoA) for the beta-diversity using the ‘bray-curtis’ distance.

```{bash, eval=FALSE}
qiime diversity beta \
--i-table table-dada2.qza \
--p-metric braycurtis \
--o-distance-matrix dada2.braycurtis.notNorm.diversity.qza
 
qiime diversity pcoa \
--i-distance-matrix dada2.braycurtis.notNorm.diversity.qza \
--o-pcoa dada2.braycurtis.notNorm.diversity.pcoa.qza
 
qiime emperor plot \
--i-pcoa dada2.braycurtis.notNorm.diversity.pcoa.qza \
--m-metadata-file metadata.file.txt \
--o-visualization dada2.braycurtis.notNorm.diversity.pcoa.qzv
```

The final `dada2.braycurtis.notNorm.diversity.pcoa.qzv` artifact contains the first 3 axes for the PCoA analysis. It is possible to colour the samples by each metadata category group, to highlight if a specific metadata may have an effect on the diversity.