# Alpha diveristy statistical analysis {#alpha_div_stats}
<center>
![](figures/alpha_div_2.png){style="width:200px; border-radius: 5px ; background: white; border: 5px solid white"}
</center>

We can carry out some statistical analysis to compare the <u id='alpha_tip'>alpha</u> diversity values between the different metadata data groups.

The QIIME2 pipeline offers two alternative choices: 

- Categorical metadata: Kruskal-Wallis test.
- Numerical metadata: Ranked Spearman (correlation) test.

## Alpha diversity: run 
<center>
![](figures/play_purple.png){style="width:150px; border-radius: 5px ; background: white; border: 5px solid white"}
</center>

As we have categorical (Location & Place) and numeric (Floor) metadata columns we will carry out both types of tests. Let's do this for the shannon <u id='alpha_tip'>alpha</u> diversity.

```{bash, eval=FALSE}
#Change directory into the output directory
cd dada2-diversity-34584
# For the Kruskal-Wallis test
qiime diversity alpha-group-significance \
--i-alpha-diversity shannon_vector.qza \
--m-metadata-file ../metadata.file.txt \
--o-visualization shannon-group-significance.kw.qzv
# For the ranked Spearman test
qiime diversity alpha-correlation \
--i-alpha-diversity shannon_vector.qza \
--m-metadata-file ../metadata.file.txt \
--o-visualization shannon-group-significance.rs.qzv
```

#### Parameters {-}
<center>
![](figures/parameter_purple.png){style="width:100px; border-radius: 5px ; background: white; border: 5px solid white"}
</center>

Although we have used two different commands they both have the same parameters.

- __`--i-alpha-diversity`__: Input <u id='artifact_tip'>artifact</u> file containing the <u id='alpha_tip'>alpha</u> diversity vectors (values).
- __`--m-metadata-file`__: [Metadata file](#metadata.file.txt) for our samples.
- __`--o-visualization`__:  Output visualisation <u id='artifact_tip'>artifact</u> containing the box plots and statistical table.

[QIIME2 docs on `qiime diversity alpha-group-significance`](https://amplicon-docs.qiime2.org/en/latest/references/plugins/diversity.html#q2-action-diversity-alpha-group-significance)

[QIIME2 docs on `qiime diversity alpha-correlation`](https://amplicon-docs.qiime2.org/en/latest/references/plugins/diversity.html#q2-action-diversity-alpha-correlation)

## Alpha diversity: visualise
<center>
![](figures/eye_purple.png){style="width:150px; border-radius: 5px ; background: white; border: 5px solid white"}
</center>

Load the produced files (`shannon-group-significance.kw.qzv` & `shannon-group-significance.rs.qzv`) into QIIME2 view using 2 chrome tabs. At this point I will remind you to have only one chrome window open which is to be left open. Also make sure to close tabs you don't need any more (always leaving one open).

At the top of both the outputs is a drop down menu labelled __"Column"__ that can be used to display the plot and stats of your chosen metadata grouping. As there was only one numeric column (Floors) there will only be one option in the Ranked Spearman output.

#### Kruskal-Wallis {-}

The Kruskal-Wallis (`shannon-group-significance.kw.qzv`) visualisation artifact contains:

- __Alpha Diversity Boxplots__: Displays the <u id='alpha_tip'>alpha</u> diversity values plotted as box plots comparing the chosen metadata column.
- __Kruskal-Wallis (all groups)__: Displays the stats for the all groups comparison. 
  - These stats will be different to the pairwise comparisons if there are more than 2 groups.
- __Kruskal-Wallis (pairwise)__: Displays the stats for the all the pairwise group comparison. 

To determine if there is a statistical difference between the groups it is best to look at the q-value/s in the pairwise table. The q-value is in essence the multiple comparison corrected p-value. As there is only one pairwise comparison in each of the 2 categorical metadata groupings the q-value is the same as the p-value (no multiple testing carried out).

[Example with more groups in the metadata columns](https://view.qiime2.org/visualization/?type=html&src=https%3A%2F%2Fdocs.qiime2.org%2F2024.10%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Ffaith-pd-group-significance.qzv)

#### Ranked Spearman {-}

The Ranked Spearman (`shannon-group-significance.rs.qzv`) visualisation artifact contains:

- __Plot__: This plots the <u id='alpha_tip'>alpha</u> diversity values as a scatterplot.
  - The x-axis contains the numeric values form the metadata.
  - The y-axis contains the <u id='alpha_tip'>alpha</u> diversity values.
- __Stats table__: The Spearman ranked correlation statistics table. 

You can look at the p-value to determine if there is a statistical significant correlation between the numeric category and the <u id='alpha_tip'>alpha</u> diversity metric.

### Alpha diversity: MCQs
<center>
![](figures/question_bubble_purple.png){style="width:150px; border-radius: 5px ; background: white; border: 5px solid white"}
</center>

```{r, echo = FALSE}
opts_p <- c("__Yes__", answer="__No__")
```
1. Is there a statistical significance (q-value <0.05) when comparing the __Locations (Corridor vs Toilet)__ when using __Shannon__ diversity? `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c("__Yes__", answer="__No__")
```
2. Is there a statistical significance (q-value <0.05) when comparing the __Places (Entrance vs MainBuilding)__ when using __Shannon__ diversity? `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c("__Yes__", answer="__No__")
```
3. Is there a statistical significance (P-value <0.05) when comparing the __Floors (1 vs 2 vs 3)__ when using __Shannon__ diversity? ? `r longmcq(opts_p)`

For the below questions you will need to run some more `qiime diversity alpha-group-significance` and `qiime diversity alpha-correlation`. When creating and running these commands ensure you are giving the input and output <u id='artifact_tip'>artifacts</u> the correct names.

If you are are having issue knowing how to run the commands you can look at the box below for the commands.

`r hide("Alpha diversity stats commands")`
```{bash, eval=FALSE}
# Q4: Observed features of Locations
qiime diversity alpha-group-significance \
--i-alpha-diversity observed_features_vector.qza \
--m-metadata-file ../metadata.file.txt \
--o-visualization observed-features-group-significance.kw.qzv
# Q5: Evenness of Locations
qiime diversity alpha-group-significance \
--i-alpha-diversity evenness_vector.qza \
--m-metadata-file ../metadata.file.txt \
--o-visualization evenness-group-significance.kw.qzv
# Q6: Faith's PD of Floors
qiime diversity alpha-correlation \
--i-alpha-diversity faith_pd_vector.qza \
--m-metadata-file ../metadata.file.txt \
--o-visualization faith-pd-group-significance.rs.qzv
```
`r unhide()`

```{r, echo = FALSE}
opts_p <- c("__Yes__", answer="__No__")
```
4. Is there a statistical significance (q-value <0.05) when comparing the __Locations (Corridor vs Toilet)__ when using __observed features__? `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c("__Yes__", answer="__No__")
```
5. Is there a statistical significance (q-value <0.05) when comparing the __Places (Entrance vs MainBuilding)__ when using __evenness__? `r longmcq(opts_p)`

```{r, echo = FALSE}
opts_p <- c("__Yes__", answer="__No__")
```
6. Is there a statistical significance (P-value <0.05) when comparing the __Floors (1 vs 2 vs 3)__ when using __Faith's PD__? `r longmcq(opts_p)`

## Alpha diversity: summary
<center>
![](figures/sum_purple.png){style="width:150px; border-radius: 5px ; background: white; border: 5px solid white"}
</center>

We have viewed <u id='alpha_tip'>alpha</u> diversity output but unfortunately found no significant differences between groups. 
Hopefully in your future projects you will have more samples and more luck. 
Even though there was no significance difference we could see some patterns in the box plots such as the diversity being higher in one group than an other.

In the next chapter we will look at the <u id='beta_tip'>beta</u> diversity output and carry out some more statistics.

```{r, echo=FALSE}
#Tippy tooltips
tippy::tippy_this(elementId = "s_tip", 
                  tooltip = "16S ribosomal RNA",
                  arrow = TRUE, placement = "bottom")
tippy::tippy_this(elementId = "otu_tip", 
                  tooltip = "Operational Taxonomy Unit",
                  arrow = TRUE, placement = "bottom")
tippy::tippy_this(elementId = "asv_tip", 
                  tooltip = "Amplicon Sequence Variant",
                  arrow = TRUE, placement = "bottom")
tippy::tippy_this(elementId = "alpha_tip", 
                  tooltip = "Alpha diversity values are based on single samples",
                  arrow = TRUE, placement = "bottom")
tippy::tippy_this(elementId = "beta_tip", 
                  tooltip = "Beta diversity values based on pairwise comparisons",
                  arrow = TRUE, placement = "bottom")
tippy::tippy_this(elementId = "artifact_tip", 
                  tooltip = "Artifact files are specific to QIIME2 and can contain data (.qza) or visualisations (.qzv)",
                  arrow = TRUE, placement = "bottom")
tippy::tippy_this(elementId = "amplicon_tip", 
                  tooltip = "Sequences created by amplifying (e.g. PCR) and sequencing (e.g. Illumina) a targeted region of DNA (e.g. 16S rRNA region)",
                  arrow = TRUE, placement = "bottom")
```