# Final consideration {#final}
```{r, fig.align = 'center',out.width= '200px', echo=FALSE }
knitr::include_graphics(path = "figures/finish.png", auto_pdf = TRUE)
``` 

Are the results you obtained with this pipeline making any sense? In a real scenario, you (as the owner of the dataset) are the one with the deepest knowledge of the dataset you are analysing, so it is upon you to answer this question. This answer needs to be inferred by gathering all the information you can on the samples, from literature or biochemical. If the answer is ‘no’, you can consider the following points.

Are there any missing taxa you know should be there? These taxa may be present but are not being detected for a few reasons, first of all, are these represented in the taxonomical database you are using? If no, you should try with a different one (Greengenes2 or Silva). If they are, you can try with different taxonomical assignment methods. If this still does not work, you can change the method to pick up the representative sequence and/or the clustering at the earlier stages of the analysis. 

The failure could be right at the beginning of the project design. It is worth checking if the PCR primers you are using can amplify these species. Are the primers known to impose bias on the community you are amplifying? Are the primers unable to amplify some taxon whose 16S rRNA sequences are too distant from the rest of the community? The literature may be able to help you on this topic.

If you are happy with the composition, you can then explore the ecological differences among your samples. For this there are few other analyses you may want to approach:

+ Inferring ecological interaction networks (SPIEC-EASI).
+ More statistical analyses with STAMP.
+ Or you can import your results in R to analyse them with Phyloseq. We have workshop on this, check our [training webpage](https://neof.org.uk/training/) to see if it is open for registration.

__CONGRATULATIONS YOU HAVE FINISHED THE ANALYSIS!__
If you have still time available you can move onto the Supplemental bookdown.