--- 
title: "Bacterial 16S metabarcoding"
author: "Luca Lenzi and Matthew Gemmell"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
favicon: figures/NEOF_favicon.png
description: NEOF book for the Bacterial 16S metabarcoding workshop
cover-image: "figures/NEOF.png"
---

```{r, fig.align = 'center',out.width= '30%', echo=FALSE }
knitr::include_graphics(path = "figures/NEOF.png", auto_pdf = TRUE)
``` 

# Introduction
```{r, fig.align = 'center',out.width= '30%', echo=FALSE }
knitr::include_graphics(path = "figures/intro_logo.png", auto_pdf = TRUE)
``` 

This practical session aims to introduce you to the analysis of bacterial 16S metabarcoding with QIIME2. The topics covered are:

- [Background on the biology](#background)
- [Introduction to QIIME2](#introqiime2)
- [Cluster and webVNC information](#cluster)
- [Information on initial data](#data)
- [QIIME2 analysis workflow](#q2workflow)
- [Sequence import](#import)
- [PCR primer trimming](#pcrtrim)
- [DADA2 denoising](#dada2)
- [Taxonomic classification](#taxa)
- [Phylogenetic tree construction](#phylogeny)
- [Rarefaction curve](#rarefaction)
- [Alpha and beta diversity analysis](#diversity)
- [Differential abundance analysis](#differential)
- [Final considerations](#final)

<a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/4.0/"><img alt="Creative Commons Licence" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/4.0/">Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License</a>.

<!--chapter:end:01-Bacterial_16S_metabarcoding.Rmd-->

# Background {#background}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/background.png", auto_pdf = TRUE)
```  
One of the most common aims of metagenomic analyses is to quantify the composition and diversity of a community (usually a community of prokaryotes). This can be done in several ways. A common approach is high-throughput amplicon sequencing, usually of a region of the bacterial 16S SSU rRNA gene. Another approach is shotgun metagenomic sequencing, where the whole genomic content of a sample is sequenced.

In the first approach, amplicons act as ‘tags’ identifying the different bacterial species present in the sample, which can be both identified and quantified to describe the composition of the bacterial community (also called _metagenetics_ or _metaprofiling_). In the second approach, whole genomes, rather than just representative amplicons, are sequenced (_shotgun metagenomics_). This DNA can be assembled, or the source of the reads can be identified and quantified in a broadly similar way to 16S analysis.

In this workshop, we will analyse amplicon sequencing data, obtained by amplification of the variable region 4 of the 16S gene, using primer 515FB-806RB (Caporaso _et al_, 2011). The sequences were obtained from the Illumina MiSeq platform, 2x250bp reads, and the analysis process will be based on the software package QIIME2 ("Quantitative Insights Into Microbial Ecology", https://qiime2.org). We will explore the bacterial populations sampled at various locations in the Bioscience building: toilets and corridors from different floors.

## 16S rRNA

16S ribosomal RNA is a structural RNA which is a component of the prokaryotic ribosome. Ribosomes are the site of protein synthesis, therefore all prokaryotes have at least a 16S rRNA gene that codes for the 16S ribosomal RNA.

The 16S rRNA gene is utilised in reconstructing phylogenies due to its slow rate of evolution, its size and its functional constancy. If the function of 16S rRNA changed then the cell could not synthesise protein correctly and so will not survive.

The 16S rRNA gene is ~1,500bp long and is made up of 9 variable regions separated by conserved regions. The variable regions evolve at a faster rate allowing differentiation between bacterial taxa. The conserved regions evolve at a slower rate which allows areas to carry out universal amplification on the gene. The nine variable regions are generally called V1, V2, V3 etc.

```{r, fig.align = 'center',out.width= '60%', echo=FALSE }
knitr::include_graphics(path = "figures/16srna.png", auto_pdf = TRUE)
```  
Secondary structure of 16S rRNA of Esherichia coli (Yarza _et al_, 2014)

## ITS and other possible marker genes
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/ITS.png", auto_pdf = TRUE)
```  
It is very common to perform analysis with marker genes other than 16S, e.g. to study communities of fungi, insects or any other eukaryote species. From the perspective of the bioinformatics analysis, the same overall pipeline can be used for most of the marker genes. 

ITS (Internal Transcribed Spacer) is a commonly used marker alternative to 16S. QIIME2 can be used to analyse ITS with a few steps and commands required which are unnecessary for 16S rRNA data. In this  tutorial, locations will be highlighted where you should consider a pipeline variation if you are using ITS. In general, we recommend searching the literature so you can be aware of the state-of-the-art methods for the analysis of your marker of interest.

## Amplicon sequencing

The use of high-throughput amplicon-sequencing has proven to be powerful and accurate for microbial community analysis. It is currently the preferred choice used to investigate biological communities (bacterial or eukaryotic) in many samples and conditions. 

Some of the reasons for this method's popularity are:

- It is relatively easy to perform.
- It can be performed with many samples.
- It is relatively fast and cheap.

On the negative side, there are many steps which may potentially introduce biases in the final result:

- Experimental design, sampling, storage and DNA extraction methods are known to potentially reduce the initial bacterial population processed and sequenced. 
- Any reagent used may add exogenous bacterial population to the samples (potentially contaminating or changing the initial bacterial amount).
- Several papers discuss how the PCR protocol used for the amplification of the DNA material may affect the final results (such as the creation of chimeric sequences or by preferentially amplifying some bacterial). 
- The choice of sequencing platform may result in the identification of different species in the same samples by having its own characteristic error-profile.
- From the bioinformatics perspective, the length of the sequences, and the pipeline used for the analysis are also able to add variability to the final results.
- The identification of the species in the samples can be affected by the quality and completeness of the database used for the analysis. Additionally, it may become less precise at lower taxonomy levels such as ‘genus’ and ‘species’ (Hugerth and Andersson, 2017).

All these aspects need to be taken into account in the discussion of the results.

```{r, fig.align = 'center',out.width= '90%', echo=FALSE }
knitr::include_graphics(path = "figures/amplicon_sequencing_representation.png", auto_pdf = TRUE)
```
Above shows a representation of amplicon sequencing.

A) Different species are shown as different colours, with the indication of which species are in the site under investigation. Some species may be excluded from the sequencing because they are:
   - Missed by the sampling methods (purple dots)
   - Lost during the sample preparation (outside green square)
   - Lost by the extraction technique (outside green square)
   - Not amplified in the PCR step (outside red square)
B) Representation of the number of sequences obtained for a sample, showing that increasing the number of sequences increases the probability to retrieve low-abundant species in the samples. The main purpose for this analysis is to infer the number and type of species in the initial sampling site by the abundance of the sequences obtained by the sequencing experiment. In order to do this effectively, the use of mock communities and negative controls is strongly suggested to evaluate the effect of any possibly introduced bias. Additionally, normalisation of the data is absolutely necessary to compare the diversity between sample groups.

<!--chapter:end:02-Background.Rmd-->

# Introduction to QIIME2 {#introqiime2}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/q2docs.png", auto_pdf = TRUE)
``` 

Since January 2018, the QIIME project released the QIIME2 (qiime2.org, Bolyen et al., 2018) pipeline to analyse amplicon sequencing data. The QIIME1 scripts, in its latest release QIIME1.9.1, are still available (and their methods are still valid) however the suggestion is to use QIIME2, as QIIME1 is no longer supported. In the following practical we will use the QIIME2 released on February 2021 (given the active development on the tool this may not be the most recent release at the time of this workshop). 

If you want to install QIIME2 on your own machine, please refer to: 
https://docs.qiime2.org/2021.2/install/

## Artifacts
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/artefact.png", auto_pdf = TRUE)
``` 

The QIIME2 pipeline produces and uses ‘artifact’ files. These contain data and metadata and may be of two types: result or visualisation. 

- __Result files (.qza)__: This file type contains the results of a method, which accepts other ‘artifacts’ and specific settings as input, in order to apply a procedure creating a new artifact (e.g. loading sequences, error correction, alpha- or beta-diversity).
- __Visualisation files (.qzv)__: This file type contains the results of a specific procedure to obtain an object that can only be used to visualize the results. These files can be loaded into the following web-site https://view.qiime2.org, where the interactive visualisations can be viewed. Please note that only chrome and Firefox are supported by this QIIME2 website.

## Workflow
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/workflow.png", auto_pdf = TRUE)
``` 

The steps for the analysis are listed below, mainly taken from https://docs.qiime2.org/2021.2/tutorials/.

```{r, fig.align = 'center',out.width= '80%', echo=FALSE }
knitr::include_graphics(path = "figures/16_tutorial_diagram.png", auto_pdf = TRUE)
``` 
Representation of the step proposed for the analysis (http://compbio.ucsd.edu/wp-content/uploads/2016/10/20170712_microbiome_16s_tutorial_non-interactive.pdf)

1. Prepare the sequence dataset
    a. Remove PCR primers
2. Amplicon Sequence Variants identification
3. Assign a taxonomic classification to each ASV
    a. Train naive-classifier for assignment
    b. Taxonomy assignment of the identified ASVs
4. Make a phylogeny tree for the ASVs 
    a. Alignment of the ASV representative sequences 
    b. Masking low quality alignment 
    c. Creating the un-rooted phylogenetic tree 
    d. Creating the rooted phylogenetic tree 
5. Summarise the taxonomy data for each sample and plot results 
6. Estimate and plot alpha diversity 
7. Estimate and plot beta diversity
8. Differential abundance analysis

## Prepare the sequence dataset
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/prepare.png", auto_pdf = TRUE)
``` 

The read set you will be using in this tutorial will be a raw dataset. It still includes low quality sequences, Illumina sequencing adapters as well as PCR primers. Before proceeding with the analysis, it is a good strategy to investigate the quality of the sequences. 

The PCR primers used for the amplification may have degenerate positions (required for annealing to the large variety of species in the samples). This adds 'random' variability outside the target region. These unknown sequences may affect the subsequent taxonomy identification as well as the error-correction step. We will use Cutadapt to remove the PCR primers before any other step. 

If the analysis is based on a marker gene different than 16S, other quality trimming may be required. In the case of the ITS marker gene, the common practice is to trim off the conserved regions (SSU, 5.8S or LSU) that may be contained in the final amplicon. The ‘ITSxpress’ software (Rivers _et al_., 2018) is designed for this specific task and it is available as a QIIME2 plug-in,  though it requires specific installation steps because it is not installed with the basic (core) installation process. It could be found among the list of third-party QIIME2 available plug-ins (https://library.qiime2.org/plugins/ ).

## Amplicon Sequence Variants identification

The key element of this type of metagenomic analysis is the identification of the different rRNA variants in the sample. 

### OTUs
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/clusters.png", auto_pdf = TRUE)
``` 

The most used procedure to identify OTUs (Operational Taxonomic Units) is through the clustering of reads into groups that are sufficiently similar to one another. These grouped reads are likely to come from the same rRNA gene, genome or taxa group. 

The identified OTUs are a feature of the sequence dataset, heavily dependent on the similarity level used for OTU identification. The most commonly used threshold is 3% dissimilarity (i.e 97% similarity) to group reads into species OTUs. Therefore, de novo OTUs identified in two different data sets cannot be compared. 

There is no single perfect tool for this step; but a few of the most widely used are:

- UCLUST (the default used by QIIME1.9.1)
- VSEARCH
- CD-Hit
- SWARM (if you work with QIIME1.9.1 we recommend using SWARM for this step)

The number of reads in each OTU should reflect the number of copies of the gene in the sample, thus providing a quantitative measure of diversity. 

### ASVs
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/mangify_glass.png", auto_pdf = TRUE)
``` 

There are now many bioinformatics methods to resolve amplicon sequence variants (ASVs) from Illumina data. These methods do not impose the arbitrary dissimilarity thresholds that define molecular OTUs. The two methods available in QIIME2 are DADA2 and deblur.

ASV methods infer the biological sequences in the sample prior to the introduction of amplification and sequencing errors, and distinguish sequence variants differing by as little as one nucleotide. The methods use a de novo process in which biological sequences are discriminated from errors on the basis of, in part, the expectation that biological sequences are more likely to be repeatedly observed than error-containing sequences.

Unlike de novo OTUs, ASVs are consistent labels because ASVs represent a biological reality that exists outside of the data being analysed: the DNA sequence of the assayed organism. Thus, ASVs inferred independently from different studies or different samples can be compared.

The following picture shows how DADA2 (an ASV method) and OTU processes compare to each other. This shows how the canonical creation of OTUs by clustering may lead to an overestimation of the size of the OTUs due to the presence of errors, from either PCR or sequencing.

```{r, fig.align = 'center',out.width= '80%', echo=FALSE }
knitr::include_graphics(path = "figures/ASC_otu_comparison.png", auto_pdf = TRUE)
``` 

Much of the work involved in analysis of amplicon-based metagenomic data is that of separating true variants from errors introduced by sequencing. These errors include miscalling of nucleotides and the generation of chimeric sequences.

For sequencing purposes, an aliquot of PhiX phage DNA is added to each sample before sequencing, both DADA2 and deblur include a step aimed to exclude any PhiX associated reads. To remove PCR chimeric artefacts, a further filter step is applied by both. 

QIIME 2 Authors suggest using DADA2 for read pairs and ‘deblur’ for already paired (stitched/joined) sequences. Please note that it is good practice to remove/trim out sequences containing Ns before the error correction step. It is not suggested to denoise with DADA2 samples from different sequencing lanes, because each lane may introduce different sequencing biases. It is possible to collate the denoised abundance tables and representative sequences afterward, recording the sequencing lane for each sample in a specific metadata column. The ‘deblur’ denoise tool may be used in this case because it should be less sensitive to these possible biases, however it is good practice to keep the lane of origin in the metadata (with the second benefit that ‘deblur’ is also faster than ‘DADA2’).

## Assign a taxonomic classification to each ASV
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/classification.png", auto_pdf = TRUE)
``` 

The sequence of each ASV is compared with the selected database, to identify the most likely taxonomy for each ASV. Unfortunately different parts of the taxonomic identification step may lead to different results for the same ASV. 

Different databases may result in different taxonomic classification for the same ASV. The choice of tool used for taxonomic assignment may also impact the results in a similar way. The current version of QIIME2 allows you to select from either the Scikit-learn classifier (a classifier which applies a machine learning approach and therefore needs training before use) or BLAST+ and VSEARCH (for a global alignment approach followed by a last common ancestor search) for this step. 

In the practical we will use Scikit-learn (similar to the RDP tool widely used so far). It has been shown that (at least in the case of 16S analysis) taxonomic classification accuracy improves when the classifier is trained with sequences derived from the amplicon region only (Werner _et al_., 2012).

## Phylogenetic tree creation
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/phylo_tree.png", auto_pdf = TRUE)
``` 

To include phylogenetic diversity metrics in the diversity analysis, such as UniFrac distance metrics, the construction of a phylogenetic tree for the identified ASVs is required. QIIME2 supports MAFFT for the alignment and MASK to mask the alignment sections that are not phylogenetically informative. Unrooted and rooted trees may be created. 

For some marker genes it is not possible to build a phylogenetic tree (ITS is among these), and it is therefore important to know the properties of the marker in use. Consequently, it is not possible to use any diversity metrics that requires the phylogenetic distance as input.

## Data normalisation
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/normalisation.png", auto_pdf = TRUE)
``` 

Any further analyses to compare the identified ASVs among sample groups (either alpha- or beta-diversity or differential abundance) should account for the features of the abundance table that can cause erroneous results (Weiss et al., 2017). These features are:

1. The microbial community in each sample may be represented by different numbers of sequences (i.e. sequencing depth for each sample).
2. The abundance table is usually sparse. That is, most of the ASVs are present only in a few samples and therefore the abundance table contains many ‘0’ values (it is also possible that a particular species is present in a sample but below the detection limit of the methods).
3. The total number of sequences obtained reflect the relative abundance of species sequenced rather that the absolute abundance of the species present in the sample (this is commonly referred as ‘compositional nature’ of the data). The relative abundances for the taxa are therefore not independent as in the underlying assumption of many frequently used statistical methods (Gloor _et al_., 2017).

A normalisation step for the data is required to mitigate these effects and to allow an easier data interpretation. The most used normalisation methods are: 

a. Normalisation by rarefaction. This is performed by reducing all the samples at the same sequence count, by random sub-sampling of the sequences.
    - This method may lead to a potential reduction of the species in your dataset (alpha-diversity) and to the exclusion of any samples not reaching the selected total count per sample.
b. DESeq method (borrowed from transcriptomic research)
c. Scaling the abundance table by a fixed value or proportion.
d. Applying a log-ratio transformation proposed by Aitchinson (1982) in analogy of compositional dataset. 
    - This is aimed to convert the abundance data into scales on which it is possible to apply conventional statistical procedures.
    
Methods b-d convert the abundance table in a normalised-abundance table with different strengths and pitfalls, therefore they can affect results for beta-diversity and differential abundant analysis.

These normalisation methods were compared by Weiss and colleagues (2017), but their simulations show that different methods may result in different sensitivity and different false discovery rates.

In this tutorial, following the QIIME2 current tutorial, we will apply the ‘rarefaction’ method, in the context of the alpha- and beta-diversity analysis.

## Alpha-diversity
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/alpha_div.png", auto_pdf = TRUE)
``` 
Alpha-diversity is used to measure the diversity within a sample. It is calculated as a value for each sample. Different metrics were developed to calculate diversity in different ways. It combines richness (a measure of the number of features in the sample) and evenness (a measure of the relative abundances of different features that make up the richness of that sample). Alpha diversity measures cannot be compared unless they have been normalised for the difference in sequencing depth between samples. The QIIME2 release we will use supports the following alpha-diversity measures, amongst others: Chao1, Shannon, Simpson, Simpson evenness, Faith’s Phylogenetic Diversity (the only one considering the phylogenetic relationships among the
ASVs).

The alpha-diversity metric used for the analysis must be chosen carefully, considering its mathematical definition in relation to the behaviour of the tools used for the previous steps. An important point to consider is whether the chosen metric uses ASVs with only 1 sequence (‘singletons’) to perform some estimation on the dataset or not. The Chao1, as one example for this, uses the number of singletons in the dataset to infer the total number of singletons in the population. However, the DADA2 denoising step excludes ‘singletons’ from the final output, making Chao1 measures not applicable to its resulting abundance tables. A second consideration regards if the phylogenetic tree is available enabling the use of the Faith’s Phylogenetic Diversity metric, for markers such as ITS this metric would not be possible.

## Beta-diversity
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/beta_div.png", auto_pdf = TRUE)
``` 
Beta-diversity is a term for the comparison of samples to each other. It is also referred as beta-diversity ordination, because it uses one (or more) technique/s to arrange samples along axes on the basis of their composition to highlight the differences. The current QIIME2 release supports only the Principal Coordinates Analysis (PCoA) method to investigate beta-diversity (within the old QIIME1 pipeline you could perform nMDS as well). The starting point of the analysis is a dissimilarity matrix, which is used to perform a Principal Component Analysis (PCA) to reduce the dimension for display purpose. The axes obtained from the PCA analysis are then compared with the categories in the metadata file to identify patterns. 

A few of the commonly used beta diversity metrics supported by QIIME2 are:

- Jaccard: A non-phylogeny based method that takes into account the presence/absence of ASVs to measure the distance between two samples.
- Bray-Curtis: A non-phylogeny based method that takes into account the number of ASVs and their abundance to determine the distance between two samples.
- Un-weighted UniFrac: Uses the presence and absence of ASVs and their phylogenies.
- Weighted UniFrac: Uses the abundance information of ASVs and their phylogenies.

As discussed for the alpha-diversity, before proceeding with the beta-diversity analysis the raw abundance table for the identified ASVs needs to be normalised. QIIME2 suggests using the rarefaction methods, in particular at the same rarefaction threshold used for the alpha diversity.

The choice of the beta-diversity distance metric to use will strongly affect the results. Therefore it is common practice to compare the results obtained with different metrics. Again, not all the metrics may be applicable to all the marker genes. E.g. the UniFrac distances could not be easily used in the case of ITS, because ITS sequences are quite difficult to align onto phylogenetic trees.

## Differential Abundance Analysis
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/comparison.png", auto_pdf = TRUE)
``` 

The identification of taxa that are differentially abundant across experimental conditions may be an important result for many types of metagenetics analyses. This step is extremely dependent on the normalization methods used to compare the samples: using different normalisation methods may lead to different results (consequently this is another active area of research).

QIIME2 supports two methods to investigate differentially abundant taxa across experimental conditions: ANCOM (Mandal et al., 2015) and GNEISS (Morton et al, 2017). Both methods are based on the conversion of the abundance values using the clr-transformation method prior to any comparison. Because the conversion is based on a logarithmic transformation, all the values in the abundance table will be increased by ‘1’ to avoid ‘0’ counts (adding pseudo-counts). These methods may therefore artificially increase the alpha-diversity for the samples as well as decrease the beta-diversity distances in the dataset. In this analysis we will use the ANCOM method following the current QIIME2 tutorial. This method computes the log-ratio for every possible pair of the identified taxa (using the mean group values for the clr-transformed abundance) and reports how many times the hypothesis “H 0 = X is not differentially abundant” is rejected for each taxa ‘X’. That is the ‘W’ statistic, the higher this value is the more significant the ASV/taxa X is differentially abundant between the tested groups. This test assumes that a large number of ASVs did not change in abundances between groups. If, in the results, more than 25% of the ASVs changes in the examined contrasts, the underlying hypothesis for the ANCOM test may be not fulfilled, and there is the chance that the results may be misleading.

Examples of questions to answer:

- Are there differences in taxa at various levels of taxonomy?
- What are the most abundant ASVs (or species)?
- What are the rare ASVs present (either real or still retained sequencing error)?
- Where are the rare ASVs present (if not sequencing error)?
- ASVs correlation: is there a correlation between ASVs and other attributes of samples, like pH or other environmental conditions?

<!--chapter:end:03-Intro_to_qiime2.Rmd-->

# Cluster Introduction {#cluster}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/cluster.png", auto_pdf = TRUE)
``` 

## Logon instructions
For this workshop we will be using Virtual Network Computing (VNC). Connect to the VNC with a browser by using the webVNC link you were sent.

You will now be in a logged-in Linux VNC desktop with two terminals. You will see something as below (there may be only one terminal which is fine). If you do not see something similar please ask for assistance.
```{r, fig.align = 'center',out.width= '80%', echo=FALSE }
knitr::include_graphics(path = "figures/logon_pic.png", auto_pdf = TRUE)
``` 

If the VNC is taking up too much/little space of your browser you can use the zoom of your browser to adjust the size. Ensure you can see one whole terminal.

These instructions will not work outside of this workshop. If you would like to install your own Linux OS on your desktop or laptop we would recommend Ubuntu. 

The following link is a guide to install Ubuntu:  
https://www.ubuntu.com/download/desktop/install-ubuntu-desktop.  
If you use a USB you need to create a bootable USB stick. The following link will assist:  
https://www.ubuntu.com/download/desktop/create-a-usb-stick-on-windows 

## The Terminal Window
In our case the terminal window looks like the picture below. We are using the terminal window as our shell to interpret our commands to the kernel. Depending on your system and preferences it may look different.
```{r, fig.align = 'center',out.width= '80%', echo=FALSE }
knitr::include_graphics(path = "figures/terminal_window.png", auto_pdf = TRUE)
``` 

Already there is useful information for us on the terminal window.

- __nsc006__: This is the login name, also known as the username. In this case nsc006 is a demonstrator's account. Your screen should show a different account name which will be your username for the Linux machine/cluster you are logged into.
- __gauss03__: This is the machine name the user is logged into.
- __\~__: This represents the current directory of the user, or the directory a command was run in. In the Linux OS and others __'~'__ is a shortcut to the user's home directory.
- Everything after the __'$'__ is where commands are typed into the terminal. This is also referred to as the command line.

__To open a new terminal window__, right click on the main screen, choose `Applications` -> `Shell` -> `bash`

<!--chapter:end:04-Cluster_Introduction.Rmd-->

# Data {#data}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/data_files.png", auto_pdf = TRUE)
``` 

Prior to starting analysis you will need three files. To acquire these three files run the following commands.

```{bash eval=FALSE}
#Change directory to home
cd ~
#Make a directory in home called "Metagenetics"
mkdir Metagenetics
#Change directory to "Metagenetics"
cd Metagenetics
#Copy the contents of a directory to your current directory
cp /pub39/tea/matthew/NEOF/16s_workshop/chapter_5/* .
```

Look at the contents of your current directory and then continue reading for explantions on their contents.

```{bash, eval=FALSE}
ls
```

## PairedEndFastqManifestPhred33.csv
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/manifest.png", auto_pdf = TRUE)
``` 

This "manifest" file contains three columns that are separated by commas (.csv = comma separated values). This is vital for importing your sequencing data into QIIME2.

Each column has a header and contains the following information:

1. __sample-id__
   - User specified sample-id. 
   - This is how the sample will be named in all downstream QIIME2 files.
   - These must match the sample ids in the metadata file.
   - Recommendations for sample ids/identifiers can be found lower down in this bookdown page.
2. __absolute-filepath__
   - Full path of sequencing file.
3. __direction__
   - Direction of the sequencing reads
   - For paired end reads this will be "forward" (R1) or "reverse" (R2).
   
For paired end read datasets there will be two lines for each sample. One for the forward read file and one for the reverse read file. These two lines must have the same sample-id.

Have a look at the "manifest" file. You will see the header followed by all the forward reads then the reverse reads.

__Tip__: Type `q` to quit `less`

```{bash, eval=FALSE}
less PairedEndFastqManifestPhred33.csv
```

This file will be used in chapter 7.

## metadata.file.txt
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/tags.png", auto_pdf = TRUE)
``` 

This file contains metadata. It consists of columns separated by tabs. 

It is vital that the "sample-id" column contains the same info as the "sample-id" column in "PairedEndFastqManifestPhred33.csv".

This file can contain as many columns as you would like (minimum of 2). The first column must be the "sample-id" column.

The first 2 rows must be:

1. Headers
   - Name of metadata fields.
   - These are chosen by the user (except the sample-id column). Make sure to these are informative and concise.
   - We suggest to only use letters and not to use spaces.
2. Column type
   - The value "#q2:types" must be used for the 1st column.
   - For all other columns the value must be:
      - "categorical": For catergorical metadata.This will be the most common choice.
      - "numeric": For numerical metadata. 
      
Have a look at the top 10 lines of the metadata file:

```{bash, eval=FALSE}
head metadata.file.txt
```
   
For more information on QIIME2 metadata please see: https://docs.qiime2.org/2021.2/tutorials/metadata/

The above link includes:

- Metadata Formatting Requirements
- Info on empy rows
- Recommendations for identifiers
- Info on column types and number formatting
- More

## primers_seqs.fasta

This is a fasta file containing the forward and reverse 16S rRNA primer sequences. These are needed for the PCR trimming step (chapter 8) and the classifier training step (chapter 10).

This file has been provided so you can copy and paste the sequences from within the webVNC. It is never advisable to try to type sequences.

For your own projects you will need to ask the lab that carried out the sequencing preparation for the primer sequences.

<!--chapter:end:05-Data.Rmd-->

# QIIME2 analysis workflow {#q2workflow}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/workflow.png", auto_pdf = TRUE)
``` 

## Workflow

The 'QIIME2' analysis workflow comprises of nine steps:

1. Import sequences into QIIME2 artifact
2. Trim the PCR primer sequences
3. De-novo amplicon sequence variants identifications
4. Assign taxonomic classification to each ASV
5. Create a phylogenetic tree of the identified ASVs
6. Sequencing depth evaluation: rarefaction plot
7. Diversity analysis: alpha and beta
8. Differential abundance analysis of ASVs across different conditions

## Activate environment
You will need to activate the QIIME2 conda environment before continuning. Carry this out with the following command.

__Note__: Ensure you include the dot and space (`. `) at the start before `useqiime2-2021.2`

```{bash eval=FALSE}
. useqiime2-2021.2
```

<!--chapter:end:06-QIIME2_analysis_workflow.Rmd-->

# (PART\*) Preprocessing of data {-}

# Sequence import {#import}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/import.png", auto_pdf = TRUE)
``` 

Now it is time to import our read pairs into a QIIME2 artifact.

__Note__: It is quite easy to make typos in the following command. Ensure to read any errors carefully to try to figure out where a typo is, if any.

```{bash, eval=FALSE}
qiime tools import --type 'SampleData[PairedEndSequencesWithQuality]' \
 --input-path PairedEndFastqManifestPhred33.csv \
 --output-path paired-end-demux.qza \
 --input-format PairedEndFastqManifestPhred33
```
 
This command will use the information contained in the ‘manifest file’ to create the ‘paired-end-demux.qza’. 

The "demux" part of the file name indicates that the samples are already demultiplexed (i.e. sequences have been separated by samples). Additionally, QIIME2 artifacts use "-" as word separators in file names.  This is not mandatory but it is a good practice to keep consistent with the QIIME2 developers.

To visualise the samples loaded, the following command will create a visualisation artifact.

```{bash, eval=FALSE}
qiime demux summarize --i-data paired-end-demux.qza \
--o-visualization paired-end-demux.qzv
```

__Chrome__

This next step is important! Make sure to open a new terminal and open chrome in it. Keep this terminal and chrome open (To open a new terminal, right click on the main screen, choose `Applications` -> `Shell` -> `bash`). Opening and closing chrome multiple times has proven to cause issues in past workshops. To open the browser ‘chrome’, run the following command in your new terminal:

```{bash, eval=FALSE}
chrome.sh
```

From now on leave the terminal you ran the above command on alone and don't close chrome.

Then:

- Navigate to the following web-site on chrome: ‘view.qiime2.org’
- Click on the ‘Drag and drop or click here’ grey box
- You can browse now to find and select the file ‘paired-end-demux.qzv’. 

__Tip__: click "HOME" on the top right, from there you can navigate to "Metagenetics" and choose your file.

The visualisation will have two tabs:

1. __Overview__: This contains 3 main sections relating to the sequence count of samples.
   - __Demultiplexxed sequence counts summary__: Summary table of number of sequences per sample divided by forward and reverse reads.
   - __Read frequency histograms__: Forward and Reverse reads histograms showing the how many samples have various numbers of sequences. 
   - __Per-sample sequence counts__: Table showing the number of forward and reverse reads in samples.
2. __Interactive Quality Plots__: Contains similar information as fastQC analysis.
   - __Interactive quality plots__: These show the quality of your forward and reverse reads using a random sampling of 10000 reads. You can click on a part of the plot to get info in the below "Paramteric seven-number summary" table.
   - __Demultiplexed sequence length summary__: Table showing summary of sequence lengths for the forward and reverse reads.

__Questions__: 

- Are the sequences evenly distributed across samples?
- Which sample has the lowest number of sequences? 
- Is there any sample with a low final number of sequences (<20k)? 
- What is the mean sequence count per sample?
- How is the overall quality for the sequences?

<!--chapter:end:07-Import_sequences_into_QIIME2.Rmd-->

# Trim the PCR primer sequences {#pcrtrim}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/prepare.png", auto_pdf = TRUE)
``` 

Reads are paired-end (R1 and R2) and may overlap in the middle, depending on the size of the sequenced amplicon and the size of the paired reads. Our sequencing data is of the 16S rRNA V4 region (length: ~250bp). Therefore, our 250bp x 2 paired end reads should overlap. If R1 and R2 overlap, we can use this to reconstruct the V4 region by aligning and merging the forward and reverse reads of each pair.

We will use the “cutadapt” (Martin, 2011) tool to remove the PCR primer sequences from R1 and R2.

__Note__: You can copy and paste the forward and reverse primer from the fasta file "primers_seqs.fasta"

```{bash, eval=FALSE}
qiime cutadapt trim-paired --i-demultiplexed-sequences paired-end-demux.qza \
 # Forward primer
 --p-front-f NNNNNGTGCCAGCMGCCGCGGTAA \
 # Reverse primer
 --p-front-r GGACTACHVGGGTWTCTAAT \
--o-trimmed-sequences paired-end-demux.trim.qza
```

To visualise information on the trimmed samples, the following command will create a visualisation artifact associated with the main sample artefact.

```{bash, eval=FALSE}
qiime demux summarize --i-data paired-end-demux.trim.qza \
--o-visualization paired-end-demux.trim.qzv
```

Now load the obtained `paired-end-demux.trim.qzv` into the ‘view.qiime2.org’ website.

__Remember to use the same chrome window!__ You can click the QIIME2 view logo on the top right of the previous `paired-end-demux.qzv` page and click "leave" on the "Do you want to leave this site?" pop-up. Now you will be on the QIIME2 view homepage.

This visualisation contains the same sections as `paired-end-demux.qzv` but contains information on the PCR trimmed data.

If you are working with the ITS region, to trim off the conserved regions from the sequences you can use the ‘q2-itxpress’ plug in. To install and use this plug-in, please see its specific manual: https://library.qiime2.org/plugins/q2-itsxpress/8/

Depending on your experimental design, you may have to perform the trimming of the PCR primer, sequencing adapters and/or using the conserved region using the ‘itsxpress’ trimming only.

__Questions__: 

- Are the sequences evenly distributed across samples?
- Which sample has the lowest number of sequences? 
- What is the average sequence count per sample? 
- How is the overall quality for the sequences? 
- Which truncating lengths would you chose for R1 and R2? This choice is based on the quality across the length of the read (looking for when quality dips) plus the total length of our V4 amplicon.

<!--chapter:end:08-Trim_PCR_sequences.Rmd-->

# (PART\*) ASVs, Taxonomy, and phylogeny {-}

# De-novo amplicon sequence variants identification {#dada2}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/clusters.png", auto_pdf = TRUE)
``` 

We will use DADA2 to identify the amplicon sequence variants (ASVs) among all the samples. 

## DADA2 denoising

To perform the variant sequence identification step with DADA2, the command is (don't run this command, see below command):

```{bash, eval=FALSE}
qiime dada2 denoise-paired \
--i-demultiplexed-seqs paired-end-demux.trim.qza \
# Truncates R1 sequences after the 220 nt length
--p-trunc-len-f 220 \
# Truncates R2 sequences after the 220 nt length
--p-trunc-len-r 220 \
# Object containing the identified sequence variants
--o-representative-sequences rep-seqs-dada2.qza \
# Object containing abundance table for the identified variants
--o-table table-dada2.qza \
--o-denoising-stats dada2-stats.qza \
--verbose
```

This step will take a long time, so stop the command (ctrl+c) and copy the final output files, which have been previously made, to your current directory:

```{bash, eval=FALSE}
#Copy the files from the specified directory your working directory
cp /pub39/tea/matthew/NEOF/16s_workshop/chapter_9/* ~/Metagenetics
```

Through the above command you will now have the following three files:

- `rep-seqs-dada2.qza`: Artifact containing the identified sequence variants.
- `tabledada2.qza`: Artifact containing the ASV abundance table. 
- `dada2-stats.qza`: Artifact which summarises the statistics after the denoising steps.

This may be one of the longest steps of the whole pipeline. The running time is proportional to the complexity of the sequences (how many ASVs are in the dataset) and to the quality of the sequences (how many errors are present). If this step is taking too long in one of your future projects, you may try to use ‘deblur’ to de-noise your data (after joining the reads with vsearch), which is generally faster.

In the case of a marker gene with very variable amplicon sequence lengths, as in the case of ITS amplicon sequencing, a possible alternative denoising strategy is to disable the filtering by truncation length using the options: `--p-trunc-len-f 0` and `--p-trunc-len-r 0`. To exclude the low quality tails the `--p-trunc-q 20` option is now required, this option will trim all the sequences after the first base with quality below ‘20’ (this may be very stringent, so you may want to try with lower values as well). A further possibility may be to use only either the forward or reverse read file, for this the `qiime dada2 denoise-single` should be used instead of the above dada2 command.

## Visualise DADA2 output
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/qiime2view.png", auto_pdf = TRUE)
``` 

### table-dada2.qzv
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/table.png", auto_pdf = TRUE)
``` 
To visualise information on the ASV table we will run:

```{bash, eval=FALSE}
qiime feature-table summarize \
--i-table table-dada2.qza \
--o-visualization table-dada2.qzv \
--m-sample-metadata-file metadata.file.txt
```
 
Now load the obtained `table-dada2.qzv` into the ‘view.qiime2.org’ website.

The page opened with `table-dada2.qzv` contains 3 tabs:

1. __Overview__: An overview page with various information.
   - __Table summary__
      - __Number of samples__
      - __Number of features__: In this case features are ASVs. If this was a taxa table features would be the number of taxa.
      - __Total frequency__: Total number of sequences (i.e. total abundance across all samples and ASVs).
   - __Frequency per sample__: Summary of number of sequences.
      - __IQR table__: Table showing the Interquartile range of sample frequencies.
      - __Histogram of frequency per sample__: This gives a rough view as the start and end of the histogram is based on the minimum and maximum sample frequencies. The vertical dark blue lines on the x axis represent the specific frequencies of samples (1 vertical line per sample).
   - __Frequency per feature__: Summary of number of sequences per feature (ASV/taxa).
      - __IQR table__: Table showing the Interquartile range of feature frequencies.
      - __Histogram of frequency per sample__: This shows a rough view of the number of features (y axis) with different frequencies (x axis). Note that the scales are logarithmic. The left side of the plot shows how many rare ASVs there are (ASVs represented by a low amount of sequences, i.e. low abundance ASVs). The right side of the plot shows the amount of high abundance ASVs.
2. __Interactive Sample Detail__: This interactive section allows you to explore sampling depth. Experiment using the Plot Controls.
   - __Plot__: This shows the number of samples per metadata category.
   - __Feature count table__: Table showing feature count per sample.
   - __Plot Controls__: Controls to change and update the plot and table.
      - __Save as..__: Controls to save the current plot.
      - __Metadata Category__: This allows you to change the values on the x axis of the plot.
      - __Sampling Depth__: Choosing a sampling depth will update the following.
         - __Plot__: The plot will be updated to grey out the tops of the bars by removing samples with a depth lower than the chosen sampling depth.
         - __Feature count table__: Samples with a lower depth than the sampling depth will be highlighted red.
         - __Plot Controls__: Below the controls is a section of text showing the number of features and samples retained by the chosen sampling depth.
3. __Feature detail__: This is a table with 3 columns.
   - 1st column: This is the ASV id. This ID represents the exact amplicon sequence variant. This is useful as if this ASV id is in another sequencing experiment it means they are identical between the two studies. This allows for easy merging of multiple ASV tables.
   - __Frequency__: Number of sequences representing the ASV acorss all samples. I.e. the total abundance of the ASV.
   - __# of Samples Observed in__: Number of samples the ASVs was detected in.

__Questions__:

- How many ASVs were detected in the dataset?
- What is the median and mean number of sequences per samples (to 1 decimal place)?
- What is the minimum and maximum number of sequences per ASV?
- How many samples have >100,000 sequences?
- Are there more Corridor samples or Toilet samples (based on Location metadata category)?
- How many sequences represent the second most abundant ASV?

### dada2-stats.qzv
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/filter.png", auto_pdf = TRUE)
``` 
Next we'll look at the denoising stats. Run the following command.

```{bash, eval=FALSE}
qiime metadata tabulate \
--m-input-file dada2-stats.qza \
--o-visualization dada2-stats.qzv
```
 
Open the `dada2-stats.qzv` visualisation with QIIME2 view.

You will now see a denoising stats table with the following columns:

- __sample-id__
- __input__: Number of sequence pairs the sample started with prior to the denoising steps.
- __filtered__: Number of sequence pairs retained after user specified filtering (e.g. `--p-trunc-len-f 220` will truncate forward reads but also remove sequence pairs where the forward length is shorter than 220).
- __percentage of input passed filter__: Percentage of input reads that passed the filtering step.
- __denoised__: Number of filtered sequence pairs that were successfully denoised.
- __merged__: Number of filtered and denoised paired sequences that were successfully merged (forward and reverse read merged into one sequence).
- __percentage of input merged__: Percentage of input reads that passed the filtering, denoising, and merging steps.
- __non-chimeric__: Number of filtered, denoised, and merged sequences retained after chimera removal. This is the final number of sequences per sample.
- __percentage of input non-chimeric__: Percentage of input reads retained at the end.

This visualisation is very useful to see where our sequences are being filtered. It is possible to be too stringent and not have enough sequences for some samples. This will give you an idea on how you can repeat the denoising stats with changed options to retain more samples.

__Questions__: 

- In which DADA2 step were most of the sequences lost?
- If this was a real study would there be enough sequences left for optimal analysis? 

<!--chapter:end:09-ASV_identification.Rmd-->

# ASV taxonomic assignment {#taxa}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/classification.png", auto_pdf = TRUE)
``` 

Now that we have defined the sequence variants in the samples, we can assign taxonomic labels to them. Here, we use the `sk-learn` classifier to assign taxonomy based on the Greengenes database (DeSantis _et al_., 2006). 

Greengenes is a database containing quality-controlled microbial sequence data. Greengenes 16S rDNA sequences come from a large number of species which have been formatted for use in QIIME, by clustering the sequences at a given level of similarity (here, 99%). As for OTU-picking, the more stringent the similarity threshold, the more fine-grained the taxonomic designation (i.e. there are more Greengenes OTUs defined to the species level at 99% than at 97% similarity). The classifier maps the variant sequences to OTU representative sequences from the Greengenes set.

## Import taxonomy database
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/import.png", auto_pdf = TRUE)
``` 

First, we need to import the Greengenes data into QIIME2 artifacts. These steps will create two objects containing the sequences and their taxonomy, respectively.

```{bash, eval=FALSE}
qiime tools import --input-path \
/pub39/tea/matthew/NEOF/16s_workshop/chapter_10/gg_13_8_otus/rep_set/99_otus.fasta \
--output-path gg-13-8.99_otus.qza \
--type 'FeatureData[Sequence]'

qiime tools import --input-path \
/pub39/tea/matthew/NEOF/16s_workshop/chapter_10/gg_13_8_otus/taxonomy/99_otu_taxonomy.txt \
--output-path gg-13-8.99.taxa.qza \
--input-format HeaderlessTSVTaxonomyFormat \
--type 'FeatureData[Taxonomy]'
```


## Extract 16s region of interest
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/scissor_cut.png", auto_pdf = TRUE)
``` 

We require a classifier training step to reach optimum assignment performance.

Prior to training we will extract our 16S region of interest. This is carried out by using the amplicon primer sequences.

Note: Do not run the below command for this tutorial, see below command.

```{bash, eval=FALSE}
qiime feature-classifier extract-reads \
--i-sequences gg-13-8.99_otus.qza \
--p-f-primer NNNNNGTGCCAGCMGCCGCGGTAA \
--p-r-primer GGACTACHVGGGTWTCTAAT \
# Creating reads of 250 bp as our intial read length was 250bp
--p-trunc-len 250 \
--o-reads gg-13-8.99.ref.seqs.qza
```

Given that the above command may require more than 14h to finish, we won’t be able to run it on this occasion. Copy over the premade results with the following command:

```{bash, eval=FALSE}
cp /pub39/tea/matthew/NEOF/16s_workshop/chapter_10/gg-13-8.99.ref.seqs.qza .
```

## Classifier training
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/train.png", auto_pdf = TRUE)
``` 

Now we train the classifier on this set of extracted reads (approx. 25 mins):

```{bash, eval=FALSE}
qiime feature-classifier fit-classifier-naive-bayes \
--i-reference-reads gg-13-8.99.ref.seqs.qza \
--i-reference-taxonomy gg-13-8.99.taxa.qza \
--o-classifier classifier.trained.qza
```

## Taxonomy assignment
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/classification.png", auto_pdf = TRUE)
``` 

With our trained classifier we can assign taxonomic assignments to the ASV representative sequences:

```{bash, eval=FALSE}
qiime feature-classifier classify-sklearn \
--i-classifier classifier.trained.qza \
# Default value for confidence
--p-confidence 0.7 \
--i-reads rep-seqs-dada2.qza \
--o-classification taxonomy.sklearn.qza
```
 
## Taxonomy bar chart production
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/bar_charts.png", auto_pdf = TRUE)
``` 

Create a taxonomy-abundance bar-chart visualisation artifact:

```{bash, eval=FALSE}
qiime taxa barplot --i-table table-dada2.qza \
--i-taxonomy taxonomy.sklearn.qza \
--m-metadata-file metadata.file.txt \
--o-visualization taxa-bar-plots.dada2.qzv
```
 
The visualisation consists of an interactive taxonomy bar chart. You can specify the taxonomic level to look at, as well as a few sorting and colouring options. Experiment with it and then answer the below questions.

__Questions__: 

- What is the most abundant taxon, at taxonomic level 5, in the dataset (note: taxa are ordered form most to least abundant in the legend)? 
- How do the samples compare to each other?

If you have sequenced any samples with known composition (e.g. mock community), now is the time to see if they behave as expected, you may want to change some steps and/or settings of what you have done so far if the composition does not match what you expect. For example, change the de-noising tool or apply more quality filters to further improve the quality of the sequence under examination.

<!--chapter:end:10-Taxonomic_assignment.Rmd-->

# Phylogenetic tree construction {#phylogeny}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/phylo_tree.png", auto_pdf = TRUE)
``` 

In this section we will obtain a phylogenetic tree of our ASV representative sequences. The final rooted tree will be used to compute phylogenetic aware alpha- and beta-diversity metrics such as Faith’s Phylogenetic Distance (Faith’s PD) and UniFrac distances. This is only carried out on marker regions for which this is possible (therefore excluding ITS among others).

## MAFFT alignment
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/dna_alignment.png", auto_pdf = TRUE)
``` 
The first step is to align our representative sequences. This is very straightforward as 16S amplicon sequences are relatively similar and short in length.

```{bash, eval=FALSE}
qiime alignment mafft \
--i-sequences rep-seqs-dada2.qza \
--o-alignment aligned-rep-seqs.qza
```

## Masking
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/mask.png", auto_pdf = TRUE)
``` 
After alignment non-informative regions are masked. This will make our phylogenetic construction quicker and better.

```{bash, eval=FALSE}
qiime alignment mask \
--i-alignment aligned-rep-seqs.qza \
--p-max-gap-frequency 1.0 \
--p-min-conservation 0.4 \
--o-masked-alignment masked-aligned-rep-seqs.qza
```

The following default options were chosen (no need to look at this it is only for your interest):

- `--p-max-gap-frequency 1.0`
   - This option is provided with a number from 0-1.
   - This determines the relative frequency of gap characters in an aligned column for the column to be retained.
   - 0.0 = Columns consisting of <=0% gaps will be retained. I.e. a column must have no gaps to be retained.
   - 1.0 = Columns consisting of <=100% gaps will be retained. I.e. no column will be removed based on number of gaps.
- `--p-min-conservation 0.4`
   - This option is provided with a number from 0-1.
   - This is the minimum relative frequency of at least one non-gap character (e.g. A,G,C,T) in a column for that column to be retained.
   - 0.4 = A column will  be retained if it contains at least one character that is present in at least 40% of the sequences. 
      - E.g. a column consisting of 25% A, 25% G, 25% C, and 25% T will __not__ be retained.
      - A column consisting of 40% A, 20% G, 20% C, and 20% T __will__ be retained.

## Un-rooted tree creation with FASTTREE
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/tree_unrooted.png", auto_pdf = TRUE)
``` 
Create an un-rooted maximum-likelihood tree from our masked alignment.

```{bash, eval=FALSE}
qiime phylogeny fasttree \
--i-alignment masked-aligned-rep-seqs.qza \
--o-tree unrooted-tree.qza
```

## Rooted tree creation
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/tree_rooted.png", auto_pdf = TRUE)
``` 
Finally the un-rooted tree is rooted.

```{bash, eval=FALSE}
qiime phylogeny midpoint-root \
--i-tree unrooted-tree.qza \
--o-rooted-tree rooted-tree.qza
```

All these steps should finish in approximately 3 minutes. The final artifact for later use is ‘rooted-tree.qza’ .

<!--chapter:end:11-Phylogentic_tree.Rmd-->

# (PART\*) Analysis {-}

# Sequencing depth evaluation: rarefaction plot {#rarefaction}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/rarefaction.png", auto_pdf = TRUE)
``` 

The detected diversity partly depends on the sampling depths. An incompletely sampled community will appear less diverse than a fully sampled community. 

To compare the diversity between samples we need to account for their different sampling depths. This can be carried out with rarefaction. 
This is performed by reducing all the samples at the same sequence count, by random sub-sampling of the sequences.

For example, if we rarefy to a depth of 2000 for all samples:

- Samples with depths of <2000 will be removed.
- Samples with depths of 2000 will be unchanged.
- Samples with depths of >2000 will have all reads removed except for a random subsample of 2000.

Equally sized subsamples are directly comparable among different samples. Additionally, we can progressively sample smaller subsamples from a sample to draw a curve of estimated alpha diversity. The shape of this curve is informative. If the curve plateaus, then increased sampling depth will not increase the estimate of alpha diversity (the community is fully sampled). 

```{r, fig.align = 'center',out.width= '80%', echo=FALSE }
knitr::include_graphics(path = "figures/rarefaction_example.png", auto_pdf = TRUE)
``` 
Example rarefaction curve: We can see that Series 1 plateaus at ~7,000 sequences and Series 3 plateaus at ~18,000 sequences whilst Series 2 has not started to plateau.

## Rarefaction command
Given an abundance table (and a phylogenetic tree for calculating phylogenetic distance), we can plot the rarefaction curve using different alpha-diversity metrics. Here, we will use diversity metrics (observed features, Simpson, Shannon, Faith’s PD) as well as an evenness metric (Simpson’s evenness).

__Note__: See the appendix at the end of this bookdown for details on each metric.

You may want to exclude `--i-phylogeny rooted-tree.qza` and `--p-metrics faith_pd` for markers such as ITS where you will not create a phylogenetic tree.

__Important__: This command will take a long time. If this is the last command you run for today that is fine, go ahead and run it.

```{bash, eval=FALSE}
qiime diversity alpha-rarefaction --i-table table-dada2.qza \
--i-phylogeny rooted-tree.qza \
# Min and max depth for the plot
# These are based on your knowledge of the sample depths
--p-min-depth 5000 \
--p-max-depth 130000 \
# Evaluating diversity every 500 sequence depth
--p-steps 500 \
# Using average from 10 iterations for the diversity count
--p-iterations 10 \
--m-metadata-file metadata.file.txt \
--p-metrics simpson_e \
--p-metrics simpson \
--p-metrics shannon \
--p-metrics observed_features \
--p-metrics faith_pd \
--o-visualization rarefaction-curve.qzv
```

Otherwise, copy over the output to your directory.

```{bash eval=FALSE}
cp /pub39/tea/matthew/NEOF/16s_workshop/chapter_12/rarefaction-curve.qzv .
```


## Rarefaction visualisation
Load the output visualisation artifact into QIIME2 view. You can select the alpha diversity __Metric__ to visualise as well as the grouping (__Sample Metadata Column__, where each __BarcodeSequence__ represents a sample). Generally the observed_features metric is the best __Metric__ to view.

The visualisation has two plots. 

- Top plot = Alpha rarefaction plot.
   - Primarily used to determine if the richness of the samples has been fully observed or sequenced.
   - If the lines in the plot appear to "plateau" (i.e., becoming a horizontal line), this suggests that collecting additional sequences beyond that sampling depth will not effect the measured metric. E.g. a plateaued observed feature curve indicates there are no more feature to be detected. 
   - If the lines in the observed feature plot do not level out:
      - The richness of the samples may have not been fully observed (because too few sequences were collected)
      - Or it could be an indicator that a lot of sequencing errors remain in the data (which is being mistaken for novel diversity).
- Bottom plot = Number of samples (Y) against sequencing depth (X)
   - This is important when grouping samples by metadata.
   - It illustrates the number of samples with different sequencing depths for the groups in the chosen __Sample Metadata Column__.
   - This is useful as when a sample's sequencing depth is lower than the sequencing depth in the rarefaction plot it will no longer be considered in the rarefaction calculation. This can be seen as jitters in the rarefaction curve corresponding to a decrease in number of samples.
      - If the sample/s had a lower value than the average of the group the curve will jitter upwards, as the group average increases. 
      - If the sample/s had a higher value than the average of the group the curve will jitter downwards, as the group average decreases. 

Choosing a rarefaction threshold for the normalisation step, is ultimately identifying the best tradeoff between:

- Retaining sequencing depth within samples.
- Losing entire samples.

The best choice usually depends on the experimental design. If you have many samples per group you may be able to lose a few samples without losing statistical power for the analysis. If you have few samples per group you may need to choose a lower threshold.

Other considerations for rarefaction threshold:

- Do the samples in one group have lower depths? Although you may be able to retain all the samples in the other groups with a high rarefaction threshold you may remove too much of the low depth group.
- Should your samples have relatively high (e.g. soil microbiomes) or low (e.g. human gut microbiome) diveristy. With higher diveristy you will need a larger rarefaction threshold.

__Questions__: 

- Are there any samples not fully sequenced? (yes or no)
- Comparing the two __Locations__, which has a higher evenness score (simpson_e)?
- Comparing the 2 __Places__, which has higher diversity value when considering most of the samples (faith_pd, shannon, & simpson)?
- What sampling depth (rarefaction threshold) would you use for the alpha- and beta-diversity analysis?

For the next step we will use a rarefaction size of 34,583. We have chosen this depth because:

- This will keep all the samples.
- The group based rarefaction curves (__Location__ and __Place__) are plateauing at this depth.
- Many of the individual samples (__BarcodeSequence__) have plateaued at this depth.
- We do not have a lot of samples so we have prioritised keeping them rather than removing samples for a relatively small gain.

How do we know that the very exact number of 34,583 will keep all the samples? Looking at the __Interactive Sample Detail__ section of the `table-dada2.qzv` visualisation we can see that the lowest depth of a sample is 34,583 (3K1E). Generally people will use the depth of a sample as the rarefaction threshold. E.g. if we were going to choose a rarefaction threshold of >50k we'd pick 58,605 as the exact threshold, this is the lowest sample depth >50k (from sample 1K2M). This allows us to use the maximum amount of sequences per sample.

<!--chapter:end:12-Rarefaction.Rmd-->

# Diversity analysis {#diversity}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/diversity.png", auto_pdf = TRUE)
``` 

Alpha diversity is a representation of the diversity within a sample. There are many possible metrics for estimating alpha diversity (QIIME2 allows 30 metrics to be estimated), including the “Shannon” metric, the number of observed features and Faith’s phylogenetic distance (Shannon,
1948, Faith, 1992).

Beta diversity is diversity among different communities. There are many possible metrics to represent the distance among samples (QIIME2 allows 34 to be estimated). We will use weighted and unweighted UniFrac metrics (Lozupone and Knight, 2005, Lozupoune et al., 2007) and the Bray-Curtis metric (Bray and Curtis, 1957) (More info on these mertics in the appendix).

## Core metrics command
```{r, fig.align = 'center',out.width= '8%', echo=FALSE }
knitr::include_graphics(path = "figures/core.png", auto_pdf = TRUE)
``` 

QIIME2 includes a wrapper script to co-investigate alpha- and beta-diversity at the same time. This script includes only a selected list of alpha- and beta-diversity metrics, but these are usually enough to investigate the dataset. It is possible to use different metrics using the single scripts for alpha and beta-diversity (https://docs.qiime2.org/2021.2/plugins/available/diversity/).

```{bash, eval=FALSE}
qiime diversity core-metrics-phylogenetic \
--i-table table-dada2.qza \
--i-phylogeny rooted-tree.qza \
# Chosen final sampling depth
--p-sampling-depth 34583 \
--output-dir dada2-diversity-34583 \
--m-metadata-file metadata.file.txt
```

This command will produce all the results within the directory specified with the option `--output-dir`. The produced artifact `rarefied_table.qza` is the abundance table normalised by rarefaction at the specified sampling size. The output directory contains artifacts for each used diversity metrics (for alpha- and beta-diversity). The beta-diversity ordination plots are also included in visualisation objects (`emperor.qzv` files).

In case of ITS or any marker for which it is not possible to obtain phylogenetic tree, the alternative command to use for the diversity analysis is: `qiime diversity core-metrics`, which lacks the
`--i-phylogeny` option.

__Questions__: 

+ Looking at the beta-diversity plot (`emperor.qzv`), one metric of your choice, is there any metadata category that produces clusters based on the groups? Note: You can reorientate the 3D plot.
+ If you choose a different metric, do you get a different plot/result?
+ Does a different sampling-depth give a different result? Try running the `core-metrics-phylogenetic` command with a different `--p-sampling-depth` and `--output-dir`.

## Alpha diveristy statistical analysis
```{r, fig.align = 'center',out.width= '10%', echo=FALSE }
knitr::include_graphics(path = "figures/alpha_div.png", auto_pdf = TRUE)
``` 

We can carry out some statistical analysis to determine if difference between the alpha diversity values in metadata groups is significantly different.

The QIIME2 pipeline offers two alternative choices: 

- Categorical data: Kruskal-Wallis test.
- Numerical data: Ranked Spearman test.

As we have catergorical and numeric metadata columns we will carry out both types of tests. Let us do this for the shannon alpha diversity.

```{bash, eval=FALSE}
#Change directory into the output directory
cd dada2-diversity-34583
# For the Kruskal-Wallis test
qiime diversity alpha-group-significance \
--i-alpha-diversity shannon_vector.qza \
--m-metadata-file ../metadata.file.txt \
--o-visualization shannon-group-significance.kw.qzv
# For the ranked Spearman test
qiime diversity alpha-correlation \
--i-alpha-diversity shannon_vector.qza \
--m-metadata-file ../metadata.file.txt \
--o-visualization shannon-group-significance.rs.qzv
```

Load the produced files into QIIME2 view. At this point I will remind you to have only one chrome window open which is to be left open. Also make sure to close tabs you don't need any more (always leaving one open).

In the visualisations you will see plots showing the diversity values and statistics below the plots. You can choose what metadata grouping to visualise with the __Column__ drop down selection. To determine if there is a statistical difference between the groups:

- Kruskal-Wallis: It is best to look at the q-value/s in the pairwise section. The q-value is in essence the multiple comparison corrected p-value. As there is only one comparison in each of the 2 categorical metadata groupings the q-value is the same as the p-value (no multiple testing carried out).
- Ranked Spearman: Can look at the p-value to determine if there is a statistical significant correlation between the numeric category and the alpha diversity metric.

__Questions__:

+ Are there any groups (for one or more metadata __Column__) showing a higher/lower alpha-diversity compared to the other/s in the same __Column__? 
+ Is the p-value or q-value for this comparison significant? 
+ If you choose a different metric do you get different results? Run both the Kruskal-Wallis and Ranked Spearman tests on another alpha diversity metric of your choice and visualise the results.

## Beta diversity statistical analysis
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/beta_div.png", auto_pdf = TRUE)
``` 

Next we can carry out statistical analysis of the beta diversity ordination plots. We will carry out a 'PERMANOVA' analysis to determine if the separation of samples by metadata groups has statistical significance or if it may have just originated by chance. 

To carry this out the 'PERMANOVA' analysis we will use the `distance_matrix.qza` which contains all the sample pairwise dissimilarity distances. For the `--p-permutations` option we will use the default value, 999.

The 'Location' metadata column with the weighted UniFrac metric will be used as there appeared to be a separation between the Corridor and Toilet samples, especially on Axis 2.

```{bash, eval=FALSE}
qiime diversity beta-group-significance \
--i-distance-matrix weighted_unifrac_distance_matrix.qza \
--m-metadata-file ../metadata.file.txt \
--m-metadata-column Location \
# prmanova default, anosim as alternative choice
--p-method permanova \
--p-pairwise \
--p-permutations 999 \
--o-visualization weighted-unifrac-significance.location.qzv
```

Load the resulting visualisation file in QIIME2 view. You will see the following sections:

- __Overview__: An overview of the statistical analysis for the metadata column.
- __Group significance plot__: These boxplots represent all the pairwise distances between the samples in the specified groups.
   - Distances to Corridor plot
      - Corridor boxplot: This boxplot represents all the pairwise distances within the Corridor samples. If you had 3 corridor samples (C1-3) the boxplot would consist of the distances between C1-C2, C1-C3, and C2-C3.
      - Toilet boxplot: This boxplot represents the pairwise distances between all the corridor samples and toilet samples. If you had 3 corridor samples (C1-3) and 2 toilet samples (T1-2) the boxplot would consist of the distances between C1-T1, C1-T2, C2-T1, C2-T2, C3-T1, and C3-T2.
   - With these plots you can see if the distances within a group is lower than the distances between that group and other groups.
- __Pairwise results__: This section shows the statistical results comparing the different groups in pairwise fashion. The best values to look at are:
   - Sample size: Good to determine if there are enough samples for statistical strength.
   - q-value: Always go for q-values over p-values.
   
For an example with more groups in the metadata column please see:
https://view.qiime2.org/visualization/?type=html&src=https%3A%2F%2Fdocs.qiime2.org%2F2021.2%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Funweighted-unifrac-body-site-significance.qzv

__Questions__:

+ Does the chosen grouping information result in a statistically significant difference? 
+ Does using ‘anosim’ instead of ‘permanova’ change the result? 
+ And using a different ordination metric? 
+ What about choosing a different grouping (metadata category)?

<!--chapter:end:13-Diversity_analysis.Rmd-->

# Differential abundance analysis {#differential}
```{r, fig.align = 'center',out.width= '30%', echo=FALSE }
knitr::include_graphics(path = "figures/comparison.png", auto_pdf = TRUE)
``` 

The ANCOM method aims to identify the differentially abundant taxa. It  is a complex method requiring different steps, for the description of the script used please refer to the script page (https://docs.qiime2.org/2021.2/plugins/available/composition/ancom/) and the ANCOM paper (https://www.ncbi.nlm.nih.gov/pubmed/26028277) .

The steps to carry this out are:

1. Collapse the ASVs into taxa
2. Add pseudo-count to the abundance table to convert ‘0’ to ‘1’
3. Run ANCOM selecting a categorical metadata to test

If you are still in ‘dada2-diversity-34583’ directory, please remember to move out:

```{bash, eval=FALSE}
cd ..
```

## Taxa collapse
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/collapse.png", auto_pdf = TRUE)
``` 

First, we need to collapse the ASVs into taxa. This is done as seeing which ASVs are differentially abundant with no taxonomic information is not very useful. Let us choose greengenes level 5 which equates to family.

```{bash, eval=FALSE}
qiime taxa collapse \
--i-table table-dada2.qza \
--i-taxonomy taxonomy.sklearn.qza \
--p-level 5 \
--o-collapsed-table comp-ancom-table-l5.qza
```

## Pseudo counts
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/plus_1.png", auto_pdf = TRUE)
``` 
Add pseudo-count to the abundance table with the purpose of converting 0s to 1s. This is required as ANCOM will fail with abundance values of 0.

```{bash, eval=FALSE}
qiime composition add-pseudocount \
--i-table comp-ancom-table-l5.qza \
--p-pseudocount 1 \
--o-composition-table comp-ancom-table-l5-pse.qza
```

## ANCOM
```{r, fig.align = 'center',out.width= '40%', echo=FALSE }
knitr::include_graphics(path = "figures/toilet_corridor.png", auto_pdf = TRUE)
``` 
Finally, we run ANCOM. In this instance we will use Location as the metadata so we can find biomarkers that differentiate between Corridor and Toilet.

```{bash, eval=FALSE}
qiime composition ancom \
--i-table comp-ancom-table-l5-pse.qza \
--m-metadata-file metadata.file.txt \
--m-metadata-column Location \
--o-visualization l5-ancom-Location.qzv
```

## ANCOM visualisation
This produces a visualisation file with the following sections:

- __ANCOM Volcano Plot__: A dot plot of the W statistic (y) against the CLR (x).
   - Each dot represents a feature (family in this case) in the dataset.
   - W statistic: The higher this value the more significant the feature is differentially abundant between groups.
   - CLR (centred log ratio): In practical terms this is how different the abundance of the feature is between the metadata groups in terms of log-fold change. The further the value is from 0 the more different the abudance.
- __ANCOM statistical results__: This shows the name and W statistic of features found to be differentially abundant.
   - ANCOM automatically chooses a W statistic threshold for significance.
- __Percentile abundances of features by group__: This is a table which can be thought of as containing the text version of a boxplot. This shows the abundances of the differentially abundant features for the different groups.


__Questions__: 

+ Which families are discovered as biomarkers? 
+ Investigating these families, does this make sense? 
+ Run ANCOM analysis for genus level (level 6). Do you get any genera as biomarkers that do not belong to families discovered as biomarkers? 
+ Does changing the metadata category let you infer something different?

<!--chapter:end:14-Differential_abundance.Rmd-->

# Final consideration {#final}
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/finish.png", auto_pdf = TRUE)
``` 

Are the results you obtained with this pipeline making any sense? In a real scenario, you (as the owner of the dataset) are the one with the deepest knowledge of the dataset you are analysing, so it is upon you to answer this question. This answer needs to be inferred by gathering all the information you can on the samples, from literature or biochemical. If the answer is ‘no’, you can consider the following points.

Are there any missing taxa you know should be there? These taxa may be present but not being detected for a few reasons, first of all, are these represented in the taxonomical database you are using? If no, you should try with a different one (Greengenes or Silva). If they are, you can try with different taxonomical assignment methods. If this still does not work, you can change the method to pick up the representative sequence and/or the clustering at the earlier stages of the analysis. 

The failure could be right at the beginning of the project design. It is worth checking if the PCR primers you are using can amplify these species. Are the primers known to impose bias on the community you are amplifying? Are the primers unable to amplify some taxon whose 16S rRNA sequences are too distant from the rest of the community? The literature may be able to help you on this topic.

If you are happy with the composition, you can then explore the ecological differences among your samples. For this there are few other analyses you may want to approach:

+ Functional analysis (PICRUSt2). We will cover this later!
+ Inferring ecological interaction networks (SPIEC-EASI)
+ More statistical analyses with STAMP
+ Or you can import your results in R to post analyse them with Phyloseq

__CONGRATULATIONS YOU HAVE FINISHED THE ANALYSIS!__
If you have still time available you can move onto the Supplemental bookdown.

<!--chapter:end:15-Final_consideration.Rmd-->

# (APPENDIX) Appendix {-}

# Resources
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/resources.png", auto_pdf = TRUE)
``` 

## Suggested readings

The QIIME2 tutorials page contains many useful protocols and possible usage for these scripts (https://docs.qiime2.org/2021.2/tutorials/).

The QIIME scripts collection (http://qiime.org/scripts/index.html) includes many other useful scripts that we have not described, each script with its own description and tips for using it. We encourage you to explore this resource.

A large resource containing links to literature on how easy it is to introduce bias into microbiome studies:

http://www.opiniomics.org/the-unbearable-madness-of-microbiome/

Diversity of 16S rRNA Genes within Individual Prokaryotic Genomes Anna Y. Pei, William E. Oberdorf, Carlos W. Nossa, Ankush Agarwal, Pooja Chokshi, Erika A. Gerz, Zhida Jin, Peng Lee, Liying Yang, Michael Poles, Stuart M. Brown, Steven Sotero, Todd DeSantis, Eoin Brodie, Karen Nelson, and Zhiheng Pei (2010) APPLIED AND ENVIRONMENTAL MICROBIOLOGY.

http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2893482/

Analysis, Optimization and Verification of Illumina- Generated 16S rRNA Gene Amplicon Surveys Michael C. Nelson, Hilary G. Morrison, Jacquelynn Benjamino, Sharon L. Grim, Joerg Graf. (2014) PlosONE

http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0094249

Interpreting 16S metagenomic data without clustering to achieve sub-OTU resolution Mikhail Tikhonov, Robert W Leach and Ned S Wingreen (2015) ISME
http://www.nature.com/ismej/journal/v9/n1/full/ismej2014117a.html

De novo clustering methods outperform reference-based methods for assigning 16S rRNA gene sequences to operational taxonomic units Sarah L. Westcott and Patrick D. Schloss (2015) PeerJ

http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4675110/

Waste Not, Want Not: Why Rarefying Microbiome Data Is Inadmissible. Paul J. McMurdie, Susan Holmes (2014) Plos ONE.

http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003531

Normalization and microbial differential abundance strategies depend upon data characteristics. Sophie Weiss, Zhenjiang Zech Xu, Shyamal Peddada, Amnon Amir, Kyle Bittinger, Antonio Gonzalez, Catherine Lozupone, Jesse R. Zaneveld, Yoshiki Vázquez-Baeza, Amanda Birmingham, Embriette R. Hyde and Rob Knight (2017) Microbiome

https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-017-0237-y

Application of multivariate statiastical techniques in microbial ecology. Paliy O, Shankar V (2016) Molecular Ecology

http://onlinelibrary.wiley.com/doi/10.1111/mec.13536/abstract

Exact sequence variants should replace operational taxonomic units in marker-gene data analysis. Benjamin J Callahan, Paul J McMurdie & Susan P Holmes (2017).

https://www.nature.com/articles/ismej2017119

Optimizing taxonomic classification of marker gene amplicon sequences Nicholas A Bokulich, Benjamin D Kaehler, Jai Ram Rideout, Matthew Dillon, Evan Bolyen, Rob Knight, Gavin A. Huttley, J. Gregory Caporaso

https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-018-0470-z

Denoising the Denoisers: An independent evaluation of microbiome 3 sequence error-correction methods. Nearing Jacob, Douglas Gavin, Comeau Andre, Langille Morngan

https://peerj.com/preprints/26566.pdf

Compositional data analysis of the microbiome: fundamentals, tools, and challenges. Tsilimigras, M.C. and Fodor, A.A.

https://www.sciencedirect.com/science/article/pii/S1047279716300722

## Other useful resources

A list of very useful site to look at for questions, suggestion and protocols

QIIME2 forum: https://forum.qiime2.org

SEQanswers: http://seqanswers.com

ResearchGate: https://www.researchgate.net/topics

## References

- Aitchison J., (1982). The statistical analysis of compositional data. Journal of the Royal Statistical Society. Series B (Methodological), pp.139-177.
- Caporaso, J. G. et al. Global patterns of 16S rRNA diversity at a depth of millions of sequences per sample. Proc Natl Acad Sci USA 108, 4516–4522 (2011)
- Gloor GB, Macklaim JM, Pawlowsky-Glahn V, and Egozcue, JJ. (2017). Microbiome datasets are compositional: and this is not optional. Frontiers in microbiology, 8, p.2224.
- Hugerth, L.W. and Andersson, A.F., 2017. Analysing microbial community composition through amplicon sequencing: from sampling to hypothesis testing. Frontiers in microbiology, 8, p.1561.
- Pollock J, Glendinning L, Wisedchanwet T, and Watson M. The madness of microbiome: Attempting to find consensus “best practice” for 16S microbiome studies. (2018) Appl Env Microbiology
- Rivers, A.R., Weber, K.C., Gardner, T.G., Liu, S. and Armstrong, S.D., 2018. ITSxpress: Software to rapidly trim internally transcribed spacer sequences with quality scores for marker gene analysis. F1000Research, 7.
- Shannon, C.E., 1948. A mathematical theory of communication. Bell system technical journal, 27(3), pp.379-423.
- Weiss S, Zech Xu ZZ, Peddada S, Amir A, Bittinger K, Gonzalez A, Lozupone C, Zaneveld JR, Vázquez-Baeza Y, Birmingham A, Hydev ER, and Knight R. Normalization and microbial differential abundance strategies depend upon data characteristics. (2017) Microbiome. 3;5(1):27
- Werner JJ, Koren O, Hugenholtz P, DeSantis TZ, Walters WA, Caporaso JG, Angenent LT, Knight R, Ley RE. Impact of training sets on classification of high-throughput bacterial 16s rRNA gene surveys. (2012) ISME J. 6(1):94-103.
- Yarza, P., Yilmaz, P., Pruesse, E., Glöckner, F.O., Ludwig, W., Schleifer, K.H., Whitman, W.B., Euzéby, J., Amann, R. and Rosselló-Móra, R., 2014. Uniting the classification of cultured and uncultured bacteria and archaea using 16S rRNA gene sequences. Nature Reviews Microbiology, 12(9), p.635.

__Cutadapt__

Martin M. Cutadapt removes adapter sequences from high-throughput sequencing reads. EMBnet.journal, 17(1):10-12, May 2011.

__QIIME2 (Quantative Insights Into Microbial Ecology)__

Bolyen E, Rideout JR, Dillon MR, Bokulich NA, Abnet C, Al-Ghalith GA, Alexander H, Alm EJ, Arumugam M, Asnicar F, Bai Y, Bisanz JE, Bittinger K, Brejnrod A, Brislawn CJ, Titus Brown C, Callahan BJ,Caraballo-Rodríguez AM, Chase J, Cope E, Da Silva R, Dorrestein PC, Douglas GM, Durall DM, Duvallet C, Edwardson CF, Ernst M, Estaki M, Fouquier J, Gauglitz JM, Gibson DL, Gonzalez A, Gorlick K, Guo J, Hillmann B, Holmes S, Holste H, Huttenhower C, Huttley G, Janssen S, Jarmusch AK, Jiang L, Kaehler B, Kang K, Keefe CR, Keim P, Kelley ST, Knights D, Koester I, Kosciolek T, Kreps J, Langille MGI, Lee J, Ley R, Liu Y, Loftfield E, Lozupone C, Maher M, Marotz C, Martin BD, McDonald D, McIver LJ, Melnik AV, Metcalf JL, Morgan SC, Morton J, Naimey AT, Navas-Molina JA, Nothias LF, Orchanian SB, Pearson T, Peoples SL, Petras D, Preuss ML, Pruesse E, Rasmussen LB, Rivers A, Robeson MS, Rosenthal P, Segata N, Shaffer M, Shiffer A, Sinha R, Song SJ, Spear JR, Swafford AD, Thompson LR, Torres PJ, Trinh P, Tripathi A, Turnbaugh PJ, Ul-Hasan S, van der Hooft JJJ, Vargas F, Vázquez-Baeza Y, Vogtmann E, von Hippel M, Walters W, Wan Y, Wang M, Warren J, Weber KC, Williamson CHD, Willis AD, Xu ZZ, Zaneveld JR, Zhang Y, Knight R, Caporaso JG (2018) QIIME 2: Reproducible, interactive, scalable, and extensible microbiome data science. PeerJ pre-prints

https://www.nature.com/articles/s41587-019-0209-9

__BLAST__

Altschul S.F., Gish W., Miller W., Myers E.W. and Lipman D.J. (1990). Basic local alignment search tool. J. Mol. Biol. 215: 403-410. 

http://blast.ncbi.nlm.nih.gov/Blast.cgi

__Bray-Curtis__

Bray, J. R. and J. T. Curtis. (1957). An ordination of upland forest communities of southern Wisconsin. Ecological Monographs 27:325-349 

__Chao1__

Chao, A. 1984. Non-parametric estimation of the number of classes in a population. Scandinavian Journal ofStatistics, 11:265-270. Colwell, R.K. and Coddington, J.A. 1994. Estimating terrestrial biodiversity through extrapolation. Philosophical Transactions of the Royal Society B: Biological Sciences, 345:101-118.

__DADA2__

Callahan BJ, McMurdie PJ, Rosen MJ, Han AW, Johnson AJA. DADA2: High-resolution sample inference from Illumina amplicon data. Nat Methods. 2016. 13, 581-583.

__GNEISS__

Morton JT, Sanders J, Quinn RA, McDonald D, Gonzalez A, Vázquez-Baeza Y, Navas-Molina JA, Song SJ, Metcalf JL, Hyde ER, Lladser M, Dorrestein PC, Knight R. (2017). Balance Trees Reveal Microbial Niche Differentiation mSystem 17;2(1)

__Greengenes__

DeSantis, T. Z., P. Hugenholtz, N. Larsen, M. Rojas, E. L. Brodie, K. Keller, T. Huber, D. Dalevi, P. Hu, and G. L. Andersen. (2006). Greengenes, a Chimera-Checked 16S rRNA Gene Database and Workbench Compatible with ARB. Appl Environ Microbiol 72:5069-72. 

__Faith’s Phylogenetic Diversity (PD)__

Faith. 1992. Conservation evaluation and phylogenetic diversity. Biological Conservation 61: 1-10

Faith DP and Baker AM. Phylogenetic diversity (PD) and biodiversity conservation: some bioinformatics challenges (2006) Evol. Bioinform. 2: 121–128.

__UniFrac__

Lozupone, C.; Knight, R. (2005). "UniFrac: A New Phylogenetic Method for Comparing Microbial Communities". Applied and Environmental Microbiology 71 (12): 8228–8235

https://aem.asm.org/content/71/12/8228.short

__UniFrac__

Lozupone, Hamady, Kelley & Knight. (2007). "Quantitative and qualitative (beta) diversity measures lead to different insights into factors that structure microbial communities." Appl Environ Microbiol 73 (5): 1576-85

__ANCOM__

Mandal S, Van Treuren W, White RA., Eggesbø M., Knight R., and Peddada, S.D. (2015). Analysis of composition of microbiomes: a novel method for studying microbial composition. Microbial ecology in health and disease, 26(1), p.27663.

<!--chapter:end:16-Resources.Rmd-->

# Diversity metrics
```{r, fig.align = 'center',out.width= '20%', echo=FALSE }
knitr::include_graphics(path = "figures/metric_square.png", auto_pdf = TRUE)
``` 

## Alpha diversity metrics
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/alpha_div.png", auto_pdf = TRUE)
``` 

Alpha diversity measures assess the diversity of each sample separately. Generally a higher value of these indexes/measures indicates higher diversity/evenness. 

### Observed features

The observed number of features is defined as the number of distinct features, such as ASVs, within a sample.

The number of observed features can also be known as the feature richness of a sample. A sample with more present features than another would be said to be richer.

### Evenness

How evenly spread the abundances are across all present features.

- A sample with 4 ASVs, each with 25% relative abundance would be perfectly even.
- A sample with 4 ASVs, where one ASV has 97% abundance and the other 3 have 1%, would be highly uneven.

### Faith's PD (phylogenetic diversity)

Faith's PD represents the minimum total branch length that covers all taxa within the sample on a phylogenetic tree (Faith, 1992). A smaller PD value indicates a reduced expected taxonomic diversity whilst a large PD value indicates a higher expected diversity.

A sample with 10 ASVs could have a lower Faith's PD than a sample with only 2 ASVs. This could occur if the 10 ASV sample only has ASVs from one genus whilst the 2 ASV samples consists of ASVs from 2 different Families.

```{r, fig.align = 'center',out.width= '60%', echo=FALSE }
knitr::include_graphics(path = "figures/faith_pd.png", auto_pdf = TRUE)
``` 

### Simpson

A measure of diversity based on number of features present and the relative abundance of each feature. If richness and evenness increase the diversity score increases.

__Equation__:
$$
D = 1 - 
{(
\frac
{\sum n{(n-1)}}
{N{(N-1)}}
)}
$$

- D = Simpson diversity index
- n = Abundance of feature
- N = Total feature abundance of sample

The values range from 0 (no diversity) to 1 (infinite diversity).

### Simpson evenness measure (simpson_e)

This is a measure of evenness based on the Simpson index. It ranges from 0 (lowest eveness) to 1 (complete evenness). It compares the calculated Simpson Index of a sample to its theoretical maximum if the sample was perfectly even but had the same amount of features.

__Equation__:
$$
E_D = 
\frac{D}
{D_m}
$$

- E~D~ = Simpson evenness measure
- D = Simpson diversity index
- D~m~ = Max possible Simpson diveristy index given the number of features

### Shannon

A measure of diversity where a higher number means higher diversity. Shannon's index accounts for both abundance and evenness of the feaures present.

__Equation__:

$$
H =
-\sum_{i=1}^{n} p_i lnp_i
$$

- H = Shannon diversity index
- p = n/N
- n = Abundance of feature
- N = Total feature abundance of sample

## Beta diversity metrics
```{r, fig.align = 'center',out.width= '15%', echo=FALSE }
knitr::include_graphics(path = "figures/beta_div.png", auto_pdf = TRUE)
``` 

Beta diversity compares 2 samples at a time. This is measured in terms of dissimilarity:

- A lower score indicate the 2 samples are more similar.
- A higher score indicates the 2 samples are more dissimilar.

### Weighted and unweighted UniFrac distances

The UniFrac metric is a phylogenetic distance measure between two samples. It is defined as “the sum of the unshared branch lengths between two samples divided by the total branch length (shared + unshared) of two samples” (Lozupone and Knight, 2005). This results in calculating the fraction of the branch lengths unique to each sample (ie. the higher this value is, the more dissimilar two samples are). The entire phylogenetic tree created for all the sequences of all the analysed samples is used for this calculation.

- Unweighted UniFrac: Only considers feature presence/absence.
- Weighted UniFrac: Takes into account feature presence/absence & abundance. I.e. the value is weighted by the abundances.

### Bray-Curtis

The Bray-Curtis metric is a dissimilarity measure that can quantify the level of difference between two samples. Two identical samples would have a Bray-Curtis measure of 0 (i.e. they have 0 dissimilarity).

There are 2 versions but the one used by QIIME2 looks at the number of features shared by the 2 samples.

__Equation__:

$$
CBC = 1 – 
(\frac{2c}
{a + b})
$$

- c = # features present in both samples
- a = # features present in sample a
- b = # features present in sample b

## Diversity Resources

For more diveristy measure please see: https://forum.qiime2.org/t/alpha-and-beta-diversity-explanations-and-commands/2282

<!--chapter:end:17-Appendix.Rmd-->

